<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Command Line Utilites &mdash; mnaseChIP 0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="mnaseChIP 0 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="index.html">mnaseChIP 0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="command-line-utilites">
<h1>Command Line Utilites<a class="headerlink" href="#command-line-utilites" title="Permalink to this headline">¶</a></h1>
<div class="section" id="convert-bam-to-bed">
<h2>Convert BAM to BED<a class="headerlink" href="#convert-bam-to-bed" title="Permalink to this headline">¶</a></h2>
<p>Transform BAM reads to resemble the true position of a ChIP seq fragment.</p>
<pre class="literal-block">
usage: bam2bed [-h] [-b BFILE] [-l LTYPE] [-e EXTEN] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file. Aligned single or paried-end ChIP-seq reads</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-l=200</span></kbd></td>
<td>INT. Library type: 0 if library is paired-end, fragment length is estimated from data; different from 0 if library is single-end, the value assigned here is used as fragment length.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-e=74</span></kbd></td>
<td>INT. Half length of each read around their fragment midpoing.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>Output file.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="counts-the-number-of-overlapping-between-two-bed-files">
<h2>Counts the number of overlapping between two BED files<a class="headerlink" href="#counts-the-number-of-overlapping-between-two-bed-files" title="Permalink to this headline">¶</a></h2>
<p>Counts the number of overlapping between two BED files.</p>
<pre class="literal-block">
usage: checkMatches [-h] [-f1 FILE1] [-f2 FILE2] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-f1</span></kbd></td>
<td>BED enrichment file replicate 1.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f2</span></kbd></td>
<td>BED enrichment file replicate 2.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>BED enrichment file with the intersection of replicates 1 and 2.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="generate-coverage-over-nucleosomes">
<h2>Generate coverage over nucleosomes<a class="headerlink" href="#generate-coverage-over-nucleosomes" title="Permalink to this headline">¶</a></h2>
<p>Generate coverage over nucleosomes.</p>
<pre class="literal-block">
usage: getAvrCov [-h] [-b BFILE] [-n NFILE] [-f FSIZE] [-w WWIDTH] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-n</span></kbd></td>
<td>BED file with enriched nucleosomes.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f</span></kbd></td>
<td>INT. Fragment size of the reads.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-w</span></kbd></td>
<td>INT. Half width of plotting window.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>Prefix of output files.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="count-reads-per-nucleosome">
<h2>Count reads per nucleosome<a class="headerlink" href="#count-reads-per-nucleosome" title="Permalink to this headline">¶</a></h2>
<p>Count reads per nucleosome.</p>
<pre class="literal-block">
usage: getCounts [-h] [-b BFILE] [-n NFILE] [-l LTYPE] [-e EXTEN] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file. ChIP-seq reads.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-n</span></kbd></td>
<td>BED file. Nucleosome data, output of Danpos. Only nucleosomes with p-values &lt; 1e-5 are used</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-l</span></kbd></td>
<td>INT. Library type: if equal to 0, data is assumed to be paired-end and read lengths are estimated from data itself; if different from 0, data is assumed to be single-end and read lengths use this value extend every read.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-e</span></kbd></td>
<td>INT. Half length of reads extended at their midpoint.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>STR. Name of output file.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="produces-coverage-figures">
<h2>Produces coverage figures<a class="headerlink" href="#produces-coverage-figures" title="Permalink to this headline">¶</a></h2>
<p>Generate coverage figures.</p>
<pre class="literal-block">
usage: getFigures [-h] [-b BFILE] [-n NFILE] [-f FRAGL] [-hr HALFR]
                  [-hw HALFW] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-n</span></kbd></td>
<td>BED file with nucleosome positions.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f</span></kbd></td>
<td>Fragment length if single-end library. If library is paired-end, provide 0 as fragment length.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-hr</span></kbd></td>
<td>Half length of reads extended at their midpoint.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-hw</span></kbd></td>
<td>Half length of the windows.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>Output file.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="id1">
<h2>Generate coverage over nucleosomes<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>Generate coverage over nucleosomes.</p>
<pre class="literal-block">
usage: getProfile [-h] [-b BFILE] [-n NFILE] [-f FSIZE] [-w WWIDTH] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-n</span></kbd></td>
<td>BED file with enriched nucleosomes.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f</span></kbd></td>
<td>INT. Fragment size of the reads.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-w</span></kbd></td>
<td>INT. Half width of plotting window.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>Prefix of output files.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="meges-the-rows-of-two-bed-files">
<h2>Meges the rows of two BED files<a class="headerlink" href="#meges-the-rows-of-two-bed-files" title="Permalink to this headline">¶</a></h2>
<p>Merge the rows of two BED files.</p>
<pre class="literal-block">
usage: merged [-h] [-f1 F1FILE] [-f2 F2FILE] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-f1</span></kbd></td>
<td>BED file. File 1.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f2</span></kbd></td>
<td>BED file. File 2.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>STR. Name of output file.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Command Line Utilites</a><ul>
<li><a class="reference internal" href="#convert-bam-to-bed">Convert BAM to BED</a></li>
<li><a class="reference internal" href="#counts-the-number-of-overlapping-between-two-bed-files">Counts the number of overlapping between two BED files</a></li>
<li><a class="reference internal" href="#generate-coverage-over-nucleosomes">Generate coverage over nucleosomes</a></li>
<li><a class="reference internal" href="#count-reads-per-nucleosome">Count reads per nucleosome</a></li>
<li><a class="reference internal" href="#produces-coverage-figures">Produces coverage figures</a></li>
<li><a class="reference internal" href="#id1">Generate coverage over nucleosomes</a></li>
<li><a class="reference internal" href="#meges-the-rows-of-two-bed-files">Meges the rows of two BED files</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/commandLineTools.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li><a href="index.html">mnaseChIP 0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, mrivas.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b2.
    </div>
  </body>
</html>