<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>A walk through nucChIP &mdash; mnaseChIP 0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="mnaseChIP 0 documentation" href="index.html" />
    <link rel="prev" title="Prerequisites and installation" href="installation.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Prerequisites and installation"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">mnaseChIP 0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="a-walk-through-nucchip">
<span id="walkthrough"></span><h1>A walk through nucChIP<a class="headerlink" href="#a-walk-through-nucchip" title="Permalink to this headline">¶</a></h1>
<p>The following walk-through assumes you have:</p>
<ol class="arabic simple">
<li>a Mnase-seq file BAM file, and</li>
<li>a set Mnase digested ChIP-seq BAM files, one per each histone mark of interest.</li>
</ol>
<div class="section" id="call-nucleosome-positions-from-mnase-data">
<h2>1. Call nucleosome positions from Mnase data<a class="headerlink" href="#call-nucleosome-positions-from-mnase-data" title="Permalink to this headline">¶</a></h2>
<p>The first step is to use Mnase-seq data to determine the nucleosome positions. We recommend to use <a class="reference external" href="http://genome.cshlp.org/content/23/2/341.short">DANPOS</a> which can be installed following these <a class="reference external" href="http://code.google.com/p/danpos/">instructions</a>.</p>
<p>Once installed, use DANPOS to call nucleosomes:</p>
<div class="highlight-python"><pre>python /path_to_danpos/danpos.py mnaseData.bam -o danpos_mnaseData</pre>
</div>
<p>DANPOS will create a folder called danpos_mnaseData where there is a file with extension <tt class="docutils literal"><span class="pre">xls</span></tt> containing the nucleosome positions along their statistics. We&#8217;ll use this on section 4, when we&#8217;ll count the number of Mnased ChIP-seq reads per nucleosome.</p>
</div>
<div class="section" id="shift-reads-on-bam-file">
<h2>2. Shift reads on BAM file<a class="headerlink" href="#shift-reads-on-bam-file" title="Permalink to this headline">¶</a></h2>
<p>We can&#8217;t use the Mnased ChIP-seq reads directly. We must correct their position to reflect their true genomic location. As shown in this figure:</p>
<img src="_images/shift_and_reLength_reads.svg" /><p>After Mnased digestion, the ChIP-seq reads may be larger or smaller than the sequence necessary to wrap the nucleosome they are attached to. In the case of paired-end data, this can be easily solved by finding the middle point between mates, and subsequent adjustment of their length. For single-end data there isn&#8217;t a direct way to find this middle point, but it can be inferred based on the typical fragment size of the library. On both cases, once determine the middle point, the true area covered by each fragment is estimated by build extending the middle point +/- 74 nt. Although +/- 74 nt is a good guest (148 is the typical length of the DNA needed to wrap a nucleosome) you may want to specify a different length. To do this seamesly you can use the nucChIP script <tt class="xref py py-class docutils literal"><span class="pre">bam2bed</span></tt>.</p>
<p>Transform BAM reads to resemble the true position of a ChIP seq fragment.</p>
<pre class="literal-block">
usage: bam2bed [-h] [-b BFILE] [-l LTYPE] [-e EXTEN] [-o OFILE]
</pre>
<dl class="docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-b</span></kbd></td>
<td>BAM file. Aligned single or paried-end ChIP-seq reads</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-l=200</span></kbd></td>
<td>INT. Library type: 0 if library is paired-end, fragment length is estimated from data; different from 0 if library is single-end, the value assigned here is used as fragment length.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-e=74</span></kbd></td>
<td>INT. Half length of each read around their fragment midpoing.</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-o</span></kbd></td>
<td>Output file.</td></tr>
</tbody>
</table>
</dd>
</dl>
</div>
<div class="section" id="plot-coverage-and-heatmaps">
<h2>3. Plot coverage and heatmaps<a class="headerlink" href="#plot-coverage-and-heatmaps" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="count-chip-seq-reads-per-nucleosome">
<h2>4. Count ChIP-seq reads per nucleosome<a class="headerlink" href="#count-chip-seq-reads-per-nucleosome" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">A walk through nucChIP</a><ul>
<li><a class="reference internal" href="#call-nucleosome-positions-from-mnase-data">1. Call nucleosome positions from Mnase data</a></li>
<li><a class="reference internal" href="#shift-reads-on-bam-file">2. Shift reads on BAM file</a></li>
<li><a class="reference internal" href="#plot-coverage-and-heatmaps">3. Plot coverage and heatmaps</a></li>
<li><a class="reference internal" href="#count-chip-seq-reads-per-nucleosome">4. Count ChIP-seq reads per nucleosome</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="installation.html"
                        title="previous chapter">Prerequisites and installation</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/walkThrough.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="installation.html" title="Prerequisites and installation"
             >previous</a> |</li>
        <li><a href="index.html">mnaseChIP 0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, mrivas.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b2.
    </div>
  </body>
</html>