<html>
<head>
<title>BibLatex Check</title>
<style>
body {
    font-family: Calibri, Arial, Sans;
    padding: 10px;
    width: 1030px;
    margin: 10 auto;
    border-top: 1px solid black;
}

#title {
    width: 720px;
    
    border-bottom: 1px solid black;
}

#title h1 {
    margin: 10px 0px;
}

#title h1 a {
    color: black;
    text-decoration: none;
}

#control {
    clear: both;
}

#search {
    float: left;
}

#search input {
    width: 300px;
    font-size: 14pt;
}

#mode {
    text-align: right;
}

#mode label:first-child {
    font-weight: bold;
}

#mode input {
    margin-left: 20px;
}

.info {
    margin-top: 10px;
    padding: 10px;
    background: #FAFADD;
    width: 250px;
    float: right;
    box-shadow: 1px 1px 1px 1px #ccc;
    clear: both;
}

.info h2 {
    font-size: 12pt;
    padding: 0px;
    margin: 0px;
}

.problem {
    margin-top: 10px;
    margin-bottom: 10px;
    padding: 10px;
    background: #FFBBAA;
    counter-increment: problem;
    width: 700px;
    border: 1px solid #993333;
    border-left: 5px solid #993333;
    box-shadow: 1px 1px 1px 1px #ccc;
    float: left;
}

.active {
    box-shadow: 5px 5px 3px 3px #ccc;
    position: relative;
    top: -2px;
}

.severe0 {
    background: #FAFAFA;
    border: 1px solid black;
    border-left: 5px solid black;
}

.severe1 {
    background: #FFEEDD;
}

.severe2 {
    background: #FFDDCC;
}

.severe3 {
    background: #FFCCBB;
}

.problem_checked {
    border: 1px solid #339933;
    border-left: 5px solid #339933;
}

.problem h2:before {
    content: counter(problem) ". "; color: gray;
}

.problem h2 {
    font-size: 12pt;
    padding: 0px;
    margin: 0px;
}

.problem .links {
    float: right;
    position:relative;
    top: -22px;
}

.problem .links a {
    color: #3333CC;
}

.problem .links a:visited {
    color: #666666;
}

.problem .reference {
    clear: both;
    font-size: 9pt;
    margin-left: 20px;
    font-style:italic;
    font-weight:bold;
}

.problem ul {
    clear: both;
}

.problem .problem_control {
    float: right;
    margin: 0px;
    padding: 0px;
}

.problem .bibtex_toggle{
    text-decoration: underline;
    font-size: 9pt;
    cursor: pointer;
    padding-top: 5px;
}

.problem .bibtex {
    margin-top: 5px;
    font-family: Monospace;
    font-size: 8pt;
    display: none;
    border: 1px solid black;
    background-color: #FFFFFF;
    padding: 5px;
}
</style>
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.5/jquery.min.js"></script>
<script>

function isInProblemMode() {
    return $('#mode_problems:checked').val() == 'problems'
}

function update() {
    $('.problem').hide();
    $('.problem[id*='+$('#search input').val()+']').show();
    $('.problem .checked').each(function () {
        if ($(this).attr('checked')) {
            $(this).parents('.problem').addClass('problem_checked');
        } else {
            $(this).parents('.problem').removeClass('problem_checked');
        }
    });
    if (isInProblemMode()) {
        $('.severe0').hide();
        $('.problem_checked').hide();
    }
}

$(document).ready(function(){

    $(".bibtex_toggle").click(function(event){
        event.preventDefault();
        $(this).next().slideToggle();
    });

    $('#search input').live('input', function() {
        update();
    });

    $('#mode input').change(function() {
        update();
    });

    $("#uncheck_button").click(function(){
        $('.problem .checked').attr('checked',false);
        localStorage.clear();
        update();
    });

    $('.problem a').mousedown(function(event) {
        $('.problem').removeClass('active');
        $(this).parents('.problem').addClass('active');
    });

    $('.problem .checked').change(function(event) {
        var problem = $(this).parents('.problem');
        problem.toggleClass('problem_checked');
        var checked = problem.hasClass('problem_checked');
        localStorage.setItem(problem.attr('id'),checked);
        if (checked && isInProblemMode()) {
            problem.slideUp();
        }
    });

    $('.problem .checked').each(function () {
        $(this).attr('checked',localStorage.getItem($(this).parents('.problem').attr('id'))=='true');
    });
    update();
});

</script>
</head>
<body>
<div id="title">
<h1><a href='http://github.com/pezmc/BibLatex-Check'>BibLaTeX Check</a></h1>
<div id="control">
<form id="search"><input placeholder="search entry ID ..."/></form>
<form id="mode">
<label>show entries:</label>
<input type = "radio"
                 name = "mode"
                 id = "mode_problems"
                 value = "problems"
                 checked = "checked" />
          <label for = "mode_problems">problems</label>
          <input type = "radio"
                 name = "mode"
                 id = "mode_all"
                 value = "all" />
          <label for = "mode_all">all</label>
<input type="button" value="uncheck all" id="uncheck_button"></button>
</form>
</div>
</div>
<div class='info'><h2>Info</h2><ul><li>bib file: Mendeley.bib</li><li>aux file: references.aux</li><li># entries: 17</li><li># problems: 1</li><ul><li># missing fields: 0</li><li># flawed names: 1</li><li># wrong types: 0</li><li># non-unique id: 0</li><li># wrong field: 0</li></ul></ul></div><div id='Allan2012' class='problem severe0'><h2>Allan2012 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Micrococcal nuclease does not substantially bias nucleosome mapping' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Micrococcal nuclease does not substantially bias nucleosome mapping' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Micrococcal nuclease does not substantially bias nucleosome mapping' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Micrococcal nuclease does not substantially bias nucleosome mapping' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Micrococcal nuclease does not substantially bias nucleosome mapping' target='_blank'>ACM</a></div><div class='reference'>Micrococcal nuclease does not substantially bias nucleosome mapping.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Allan2012,<br />abstract = {We have mapped sequence-directed nucleosome positioning on genomic DNA molecules using high-throughput sequencing. Chromatins, prepared by reconstitution with either chicken or frog histones, were separately digested to mononucleosomes using either micrococcal nuclease (MNase) or caspase-activated DNase (CAD). Both enzymes preferentially cleave internucleosomal (linker) DNA, although they do so by markedly different mechanisms. MNase has hitherto been very widely used to map nucleosomes, although concerns have been raised over its potential to introduce bias. Having identified the locations and quantified the strength of both the chicken or frog histone octamer binding sites on each DNA, the results obtained with the two enzymes were compared using a variety of criteria. Both enzymes displayed sequence specificity in their preferred cleavage sites, although the nature of this selectivity was distinct for the two enzymes. In addition, nucleosomes produced by CAD nuclease are 8-10 bp longer than those produced with MNase, with the CAD cleavage sites tending to be 4-5 bp further out from the nucleosomal dyad than the corresponding MNase cleavage sites. Despite these notable differences in cleavage behaviour, the two nucleases identified essentially equivalent patterns of nucleosome positioning sites on each of the DNAs tested, an observation that was independent of the histone type. These results indicate that biases in nucleosome positioning data collected using MNase are, under our conditions, not significant.},<br />author = {Allan, James and Fraser, Ross M and Owen-Hughes, Tom and Keszenman-Pereyra, David},<br />doi = {10.1016/j.jmb.2012.01.043},<br />issn = {1089-8638},<br />journaltitle = {Journal of molecular biology},<br />keywords = {Animals,Binding Sites,Chickens,Chickens: genetics,Chromatin,Chromatin: metabolism,Deoxyribonucleases,Deoxyribonucleases: chemistry,Deoxyribonucleases: metabolism,Histones,Histones: genetics,Histones: metabolism,Lactoglobulins,MNase is not sequence bias,Micrococcal Nuclease,Micrococcal Nuclease: chemistry,Micrococcal Nuclease: metabolism,Models, Molecular,Nucleosomes,Nucleosomes: genetics,Nucleosomes: metabolism,Protein Conformation,Ranidae,Ranidae: genetics,Restriction Mapping,Restriction Mapping: methods},<br />mendeley-tags = {MNase is not sequence bias},<br />month = mar,<br />number = {3},<br />pages = {152--64},<br />pmid = {22310051},<br />title = {{Micrococcal nuclease does not substantially bias nucleosome mapping.}},<br />url = {http://www.sciencedirect.com/science/article/pii/S0022283612000976},<br />volume = {417},<br />year = {2012}<br />}<br /></div></div><div id='Barski2007' class='problem severe0'><h2>Barski2007 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Highresolution profiling of histone methylations in the human genome' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Highresolution profiling of histone methylations in the human genome' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Highresolution profiling of histone methylations in the human genome' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Highresolution profiling of histone methylations in the human genome' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Highresolution profiling of histone methylations in the human genome' target='_blank'>ACM</a></div><div class='reference'>High-resolution profiling of histone methylations in the human genome.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Barski2007,<br />abstract = {Histone modifications are implicated in influencing gene expression. We have generated high-resolution maps for the genome-wide distribution of 20 histone lysine and arginine methylations as well as histone variant H2A.Z, RNA polymerase II, and the insulator binding protein CTCF across the human genome using the Solexa 1G sequencing technology. Typical patterns of histone methylations exhibited at promoters, insulators, enhancers, and transcribed regions are identified. The monomethylations of H3K27, H3K9, H4K20, H3K79, and H2BK5 are all linked to gene activation, whereas trimethylations of H3K27, H3K9, and H3K79 are linked to repression. H2A.Z associates with functional regulatory elements, and CTCF marks boundaries of histone methylation domains. Chromosome banding patterns are correlated with unique patterns of histone modifications. Chromosome breakpoints detected in T cell cancers frequently reside in chromatin regions associated with H3K4 methylations. Our data provide new insights into the function of histone methylation and chromatin organization in genome function.},<br />author = {Barski, Artem and Cuddapah, Suresh and Cui, Kairong and Roh, Tae-Young and Schones, Dustin E and Wang, Zhibin and Wei, Gang and Chepelev, Iouri and Zhao, Keji},<br />doi = {10.1016/j.cell.2007.05.009},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Barski et al. - 2007 - High-resolution profiling of histone methylations in the human genome.pdf:pdf},<br />issn = {0092-8674},<br />journaltitle = {Cell},<br />keywords = {Chromatin,Chromatin: genetics,Chromatin: ultrastructure,Chromosome Breakage,Enhancer Elements, Genetic,Enhancer Elements, Genetic: genetics,Epigenesis, Genetic,Epigenesis, Genetic: genetics,Gene Expression Profiling,Gene Expression Profiling: methods,Gene Expression Regulation,Gene Expression Regulation: genetics,Genome, Human,Genome, Human: genetics,H3K9me3 and H3K27me3 as repressors,Histone-Lysine N-Methyltransferase,Histone-Lysine N-Methyltransferase: metabolism,Histones,Histones: genetics,Histones: metabolism,Humans,Lymphoma,Lymphoma: genetics,Methylation,Promoter Regions, Genetic,Promoter Regions, Genetic: genetics,Protein Methyltransferases,RNA Polymerase II,RNA Polymerase II: metabolism,Regulatory Elements, Transcriptional,Regulatory Elements, Transcriptional: genetics,Transcriptional Activation,Transcriptional Activation: genetics},<br />language = {English},<br />mendeley-tags = {H3K9me3 and H3K27me3 as repressors},<br />month = may,<br />number = {4},<br />pages = {823--37},<br />pmid = {17512414},<br />publisher = {Elsevier},<br />title = {{High-resolution profiling of histone methylations in the human genome.}},<br />url = {http://www.cell.com/article/S0092867407006009/fulltext},<br />volume = {129},<br />year = {2007}<br />}<br /></div></div><div id='Carone2014' class='problem severe0'><h2>Carone2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=HighResolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=HighResolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=HighResolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=HighResolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=HighResolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm' target='_blank'>ACM</a></div><div class='reference'>High-Resolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Carone2014,<br />abstract = {Mammalian embryonic stem cells (ESCs) and sperm exhibit unusual chromatin packaging that plays important roles in cellular function. Here, we extend a recently developed technique, based on deep paired-end sequencing of lightly digested chromatin, to assess footprints of nucleosomes and other DNA-binding proteins genome-wide in murine ESCs and sperm. In ESCs, we recover well-characterized features of chromatin such as promoter nucleosome depletion and further identify widespread footprints of sequence-specific DNA-binding proteins such as CTCF, which we validate in knockdown studies. We document global differences in nuclease accessibility between ESCs and sperm, finding that the majority of histone retention in sperm preferentially occurs in large gene-poor genomic regions, with only a small subset of nucleosomes being retained over promoters of developmental regulators. Finally, we describe evidence that CTCF remains associated with the genome in mature sperm, where it could play a role in organizing the sperm genome.},<br />annote = {Paired-end library of MNase data at E14},<br />author = {Carone, Benjamin R and Hung, Jui-Hung and Hainer, Sarah J and Chou, Min-Te and Carone, Dawn M and Weng, Zhiping and Fazzio, Thomas G and Rando, Oliver J},<br />doi = {10.1016/j.devcel.2014.05.024},<br />issn = {1878-1551},<br />journaltitle = {Developmental cell},<br />keywords = {8\_mnase,Animals,Cells, Cultured,Chromatin,Chromatin Immunoprecipitation,Chromatin: genetics,Chromatin: metabolism,Chromosome Mapping,Chromosome Mapping: methods,DNA Footprinting,Embryonic Stem Cells,Embryonic Stem Cells: cytology,Embryonic Stem Cells: metabolism,Genome,High-Throughput Nucleotide Sequencing,Histones,Histones: metabolism,MNase on E14,Male,Mice,Nucleosomes,Nucleosomes: metabolism,Promoter Regions, Genetic,Promoter Regions, Genetic: genetics,Repressor Proteins,Repressor Proteins: metabolism,Spermatozoa,Spermatozoa: metabolism,Transcription Factors,Transcription Factors: genetics,Transcription Factors: metabolism,Transcription, Genetic},<br />language = {English},<br />mendeley-tags = {8\_mnase,MNase on E14},<br />month = jul,<br />number = {1},<br />pages = {11--22},<br />pmid = {24998598},<br />publisher = {Elsevier},<br />title = {{High-Resolution Mapping of Chromatin Packaging in Mouse Embryonic Stem Cells and Sperm.}},<br />url = {http://www.cell.com/article/S153458071400344X/fulltext},<br />volume = {30},<br />year = {2014}<br />}<br /></div></div><div id='Chen2014' class='problem severe0'><h2>Chen2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Improved nucleosomepositioning algorithm iNPS for accurate nucleosome positioning from sequencing data' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Improved nucleosomepositioning algorithm iNPS for accurate nucleosome positioning from sequencing data' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Improved nucleosomepositioning algorithm iNPS for accurate nucleosome positioning from sequencing data' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Improved nucleosomepositioning algorithm iNPS for accurate nucleosome positioning from sequencing data' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Improved nucleosomepositioning algorithm iNPS for accurate nucleosome positioning from sequencing data' target='_blank'>ACM</a></div><div class='reference'>Improved nucleosome-positioning algorithm iNPS for accurate nucleosome positioning from sequencing data.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Chen2014,<br />abstract = {Accurate determination of genome-wide nucleosome positioning can provide important insights into global gene regulation. Here, we describe the development of an improved nucleosome-positioning algorithm-iNPS-which achieves significantly better performance than the widely used NPS package. By determining nucleosome boundaries more precisely and merging or separating shoulder peaks based on local MNase-seq signals, iNPS can unambiguously detect 60\% more nucleosomes. The detected nucleosomes display better nucleosome 'widths' and neighbouring centre-centre distance distributions, giving rise to sharper patterns and better phasing of average nucleosome profiles and higher consistency between independent data subsets. In addition to its unique advantage in classifying nucleosomes by shape to reveal their different biological properties, iNPS also achieves higher significance and lower false positive rates than previously published methods. The application of iNPS to T-cell activation data demonstrates a greater ability to facilitate detection of nucleosome repositioning, uncovering additional biological features underlying the activation process.},<br />author = {Chen, Weizhong and Liu, Yi and Zhu, Shanshan and Green, Christopher D and Wei, Gang and Han, Jing-Dong Jackie},<br />doi = {10.1038/ncomms5909},<br />isbn = {doi:10.1038/ncomms5909},<br />issn = {2041-1723},<br />journaltitle = {Nature communications},<br />language = {en},<br />month = jan,<br />pages = {4909},<br />pmid = {25233085},<br />publisher = {Nature Publishing Group},<br />title = {{Improved nucleosome-positioning algorithm iNPS for accurate nucleosome positioning from sequencing data.}},<br />url = {http://www.nature.com/ncomms/2014/140918/ncomms5909/full/ncomms5909.html},<br />volume = {5},<br />year = {2014}<br />}<br /></div></div><div id='Haberle2014' class='problem severe0'><h2>Haberle2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Two independent transcription initiation codes overlap on vertebrate core promoters' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Two independent transcription initiation codes overlap on vertebrate core promoters' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Two independent transcription initiation codes overlap on vertebrate core promoters' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Two independent transcription initiation codes overlap on vertebrate core promoters' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Two independent transcription initiation codes overlap on vertebrate core promoters' target='_blank'>ACM</a></div><div class='reference'>Two independent transcription initiation codes overlap on vertebrate core promoters.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Haberle2014,<br />abstract = {A core promoter is a stretch of DNA surrounding the transcription start site (TSS) that integrates regulatory inputs and recruits general transcription factors to initiate transcription. The nature and causative relationship of the DNA sequence and chromatin signals that govern the selection of most TSSs by RNA polymerase II remain unresolved. Maternal to zygotic transition represents the most marked change of the transcriptome repertoire in the vertebrate life cycle. Early embryonic development in zebrafish is characterized by a series of transcriptionally silent cell cycles regulated by inherited maternal gene products: zygotic genome activation commences at the tenth cell cycle, marking the mid-blastula transition. This transition provides a unique opportunity to study the rules of TSS selection and the hierarchy of events linking transcription initiation with key chromatin modifications. We analysed TSS usage during zebrafish early embryonic development at high resolution using cap analysis of gene expression, and determined the positions of H3K4me3-marked promoter-associated nucleosomes. Here we show that the transition from the maternal to zygotic transcriptome is characterized by a switch between two fundamentally different modes of defining transcription initiation, which drive the dynamic change of TSS usage and promoter shape. A maternal-specific TSS selection, which requires an A/T-rich (W-box) motif, is replaced with a zygotic TSS selection grammar characterized by broader patterns of dinucleotide enrichments, precisely aligned with the first downstream (+1) nucleosome. The developmental dynamics of the H3K4me3-marked nucleosomes reveal their DNA-sequence-associated positioning at promoters before zygotic transcription and subsequent transcription-independent adjustment to the final position downstream of the zygotic TSS. The two TSS-defining grammars coexist, often physically overlapping, in core promoters of constitutively expressed genes to enable their expression in the two regulatory environments. The dissection of overlapping core promoter determinants represents a framework for future studies of promoter structure and function across different regulatory contexts.},<br />author = {Haberle, Vanja and Li, Nan and Hadzhiev, Yavor and Plessy, Charles and Previti, Christopher and Nepal, Chirag and Gehrig, Jochen and Dong, Xianjun and Akalin, Altuna and Suzuki, Ana Maria and van IJcken, Wilfred F J and Armant, Olivier and Ferg, Marco and Str\"{a}hle, Uwe and Carninci, Piero and M\"{u}ller, Ferenc and Lenhard, Boris},<br />doi = {10.1038/nature12974},<br />issn = {1476-4687},<br />journaltitle = {Nature},<br />keywords = {Animals,Base Sequence,Embryo, Nonmammalian,Embryo, Nonmammalian: embryology,Embryo, Nonmammalian: metabolism,Female,Gene Expression Regulation, Developmental,Gene Expression Regulation, Developmental: genetic,Histones,Histones: metabolism,Methylation,Mothers,Nucleosomes,Nucleosomes: genetics,Promoter Regions, Genetic,Promoter Regions, Genetic: genetics,Sequence based code for transcription,Transcription Initiation Site,Transcription Initiation, Genetic,Transcriptome,Transcriptome: genetics,Zebrafish,Zebrafish: embryology,Zebrafish: genetics,Zygote,Zygote: metabolism},<br />mendeley-tags = {Sequence based code for transcription},<br />month = mar,<br />number = {7492},<br />pages = {381--5},<br />pmid = {24531765},<br />publisher = {Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.},<br />shorttitle = {Nature},<br />title = {{Two independent transcription initiation codes overlap on vertebrate core promoters.}},<br />url = {http://dx.doi.org/10.1038/nature12974},<br />volume = {507},<br />year = {2014}<br />}<br /></div></div><div id='Jiang2009' class='problem severe1'><h2>Jiang2009 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Nucleosome positioning and gene regulation advances through genomics' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Nucleosome positioning and gene regulation advances through genomics' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Nucleosome positioning and gene regulation advances through genomics' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Nucleosome positioning and gene regulation advances through genomics' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Nucleosome positioning and gene regulation advances through genomics' target='_blank'>ACM</a></div><div class='reference'>Nucleosome positioning and gene regulation: advances through genomics.</div><ul><li>flawed name: abbreviated journal title 'Nature reviews. Genetics'</li></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Jiang2009,<br />abstract = {Knowing the precise locations of nucleosomes in a genome is key to understanding how genes are regulated. Recent 'next generation' ChIP-chip and ChIP-Seq technologies have accelerated our understanding of the basic principles of chromatin organization. Here we discuss what high-resolution genome-wide maps of nucleosome positions have taught us about how nucleosome positioning demarcates promoter regions and transcriptional start sites, and how the composition and structure of promoter nucleosomes facilitate or inhibit transcription. A detailed picture is starting to emerge of how diverse factors, including underlying DNA sequences and chromatin remodelling complexes, influence nucleosome positioning.},<br />author = {Jiang, Cizhong and Pugh, B Franklin},<br />doi = {10.1038/nrg2522},<br />issn = {1471-0064},<br />journaltitle = {Nature reviews. Genetics},<br />keywords = {Animals,Chromatin Assembly and Disassembly,Gene Expression Regulation,Genomics,Humans,Linker DNA \~{}38 nt,Nucleosomes,Nucleosomes: metabolism,Saccharomyces cerevisiae,Saccharomyces cerevisiae: genetics,Saccharomyces cerevisiae: metabolism},<br />mendeley-tags = {Linker DNA \~{}38 nt},<br />month = mar,<br />number = {3},<br />pages = {161--72},<br />pmid = {19204718},<br />publisher = {Nature Publishing Group},<br />shorttitle = {Nat Rev Genet},<br />title = {{Nucleosome positioning and gene regulation: advances through genomics.}},<br />url = {http://dx.doi.org/10.1038/nrg2522},<br />volume = {10},<br />year = {2009}<br />}<br /></div></div><div id='Kaplan2009' class='problem severe0'><h2>Kaplan2009 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=The DNAencoded nucleosome organization of a eukaryotic genome' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=The DNAencoded nucleosome organization of a eukaryotic genome' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=The DNAencoded nucleosome organization of a eukaryotic genome' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=The DNAencoded nucleosome organization of a eukaryotic genome' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=The DNAencoded nucleosome organization of a eukaryotic genome' target='_blank'>ACM</a></div><div class='reference'>The DNA-encoded nucleosome organization of a eukaryotic genome.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Kaplan2009,<br />abstract = {Nucleosome organization is critical for gene regulation. In living cells this organization is determined by multiple factors, including the action of chromatin remodellers, competition with site-specific DNA-binding proteins, and the DNA sequence preferences of the nucleosomes themselves. However, it has been difficult to estimate the relative importance of each of these mechanisms in vivo, because in vivo nucleosome maps reflect the combined action of all influencing factors. Here we determine the importance of nucleosome DNA sequence preferences experimentally by measuring the genome-wide occupancy of nucleosomes assembled on purified yeast genomic DNA. The resulting map, in which nucleosome occupancy is governed only by the intrinsic sequence preferences of nucleosomes, is similar to in vivo nucleosome maps generated in three different growth conditions. In vitro, nucleosome depletion is evident at many transcription factor binding sites and around gene start and end sites, indicating that nucleosome depletion at these sites in vivo is partly encoded in the genome. We confirm these results with a micrococcal nuclease-independent experiment that measures the relative affinity of nucleosomes for approximately 40,000 double-stranded 150-base-pair oligonucleotides. Using our in vitro data, we devise a computational model of nucleosome sequence preferences that is significantly correlated with in vivo nucleosome occupancy in Caenorhabditis elegans. Our results indicate that the intrinsic DNA sequence preferences of nucleosomes have a central role in determining the organization of nucleosomes in vivo.},<br />author = {Kaplan, Noam and Moore, Irene K and Fondufe-Mittendorf, Yvonne and Gossett, Andrea J and Tillo, Desiree and Field, Yair and LeProust, Emily M and Hughes, Timothy R and Lieb, Jason D and Widom, Jonathan and Segal, Eran},<br />doi = {10.1038/nature07667},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Kaplan et al. - 2009 - The DNA-encoded nucleosome organization of a eukaryotic genome.pdf:pdf},<br />issn = {1476-4687},<br />journaltitle = {Nature},<br />keywords = {Animals,Base Sequence,Caenorhabditis elegans,Caenorhabditis elegans: genetics,Chickens,Computational Biology,Computer Simulation,DNA leads nucleosome positioning,Eukaryotic Cells,Eukaryotic Cells: metabolism,Genome, Fungal,Genome, Fungal: genetics,Micrococcal Nuclease,Micrococcal Nuclease: metabolism,Nucleosomes,Nucleosomes: genetics,Nucleosomes: metabolism,RNA, Messenger,RNA, Messenger: genetics,RNA, Messenger: metabolism,Saccharomyces cerevisiae,Saccharomyces cerevisiae: genetics,Saccharomyces cerevisiae: growth \& development,Sequence Analysis, DNA,Transcription Factors,Transcription Factors: metabolism},<br />mendeley-tags = {DNA leads nucleosome positioning},<br />month = mar,<br />number = {7236},<br />pages = {362--6},<br />pmid = {19092803},<br />publisher = {Macmillan Publishers Limited. All rights reserved},<br />shorttitle = {Nature},<br />title = {{The DNA-encoded nucleosome organization of a eukaryotic genome.}},<br />url = {http://dx.doi.org/10.1038/nature07667},<br />volume = {458},<br />year = {2009}<br />}<br /></div></div><div id='Koch2007' class='problem severe0'><h2>Koch2007 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=The landscape of histone modifications across 1 of the human genome in five human cell lines' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=The landscape of histone modifications across 1 of the human genome in five human cell lines' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=The landscape of histone modifications across 1 of the human genome in five human cell lines' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=The landscape of histone modifications across 1 of the human genome in five human cell lines' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=The landscape of histone modifications across 1 of the human genome in five human cell lines' target='_blank'>ACM</a></div><div class='reference'>The landscape of histone modifications across 1\% of the human genome in five human cell lines.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Koch2007,<br />abstract = {We generated high-resolution maps of histone H3 lysine 9/14 acetylation (H3ac), histone H4 lysine 5/8/12/16 acetylation (H4ac), and histone H3 at lysine 4 mono-, di-, and trimethylation (H3K4me1, H3K4me2, H3K4me3, respectively) across the ENCODE regions. Studying each modification in five human cell lines including the ENCODE Consortium common cell lines GM06990 (lymphoblastoid) and HeLa-S3, as well as K562, HFL-1, and MOLT4, we identified clear patterns of histone modification profiles with respect to genomic features. H3K4me3, H3K4me2, and H3ac modifications are tightly associated with the transcriptional start sites (TSSs) of genes, while H3K4me1 and H4ac have more widespread distributions. TSSs reveal characteristic patterns of both types of modification present and the position relative to TSSs. These patterns differ between active and inactive genes and in particular the state of H3K4me3 and H3ac modifications is highly predictive of gene activity. Away from TSSs, modification sites are enriched in H3K4me1 and relatively depleted in H3K4me3 and H3ac. Comparison between cell lines identified differences in the histone modification profiles associated with transcriptional differences between the cell lines. These results provide an overview of the functional relationship among histone modifications and gene expression in human cells.},<br />author = {Koch, Christoph M and Andrews, Robert M and Flicek, Paul and Dillon, Shane C and Kara\"{o}z, Ula≈ü and Clelland, Gayle K and Wilcox, Sarah and Beare, David M and Fowler, Joanna C and Couttet, Phillippe and James, Keith D and Lefebvre, Gregory C and Bruce, Alexander W and Dovey, Oliver M and Ellis, Peter D and Dhami, Pawandeep and Langford, Cordelia F and Weng, Zhiping and Birney, Ewan and Carter, Nigel P and Vetrie, David and Dunham, Ian},<br />doi = {10.1101/gr.5704207},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Koch et al. - 2007 - The landscape of histone modifications across 1\% of the human genome in five human cell lines.pdf:pdf},<br />issn = {1088-9051},<br />journaltitle = {Genome research},<br />keywords = {Genome, Human,Genome, Human: physiology,H3K4me3 as activator of gene expression,HeLa Cells,Histones,Histones: metabolism,Humans,K562 Cells,Protein Processing, Post-Translational,Protein Processing, Post-Translational: physiology,Transcription, Genetic,Transcription, Genetic: physiology},<br />mendeley-tags = {H3K4me3 as activator of gene expression},<br />month = jun,<br />number = {6},<br />pages = {691--707},<br />pmid = {17567990},<br />title = {{The landscape of histone modifications across 1\% of the human genome in five human cell lines.}},<br />url = {http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=1891331\&tool=pmcentrez\&rendertype=abstract},<br />volume = {17},<br />year = {2007}<br />}<br /></div></div><div id='Luco2011' class='problem severe0'><h2>Luco2011 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Epigenetics in alternative premRNA splicing' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Epigenetics in alternative premRNA splicing' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Epigenetics in alternative premRNA splicing' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Epigenetics in alternative premRNA splicing' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Epigenetics in alternative premRNA splicing' target='_blank'>ACM</a></div><div class='reference'>Epigenetics in alternative pre-mRNA splicing.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Luco2011,<br />abstract = {Alternative splicing plays critical roles in differentiation, development, and disease and is a major source for protein diversity in higher eukaryotes. Analysis of alternative splicing regulation has traditionally focused on RNA sequence elements and their associated splicing factors, but recent provocative studies point to a key function of chromatin structure and histone modifications in alternative splicing regulation. These insights suggest that epigenetic regulation determines not only what parts of the genome are expressed but also how they are spliced.},<br />author = {Luco, Reini F and Allo, Mariano and Schor, Ignacio E and Kornblihtt, Alberto R and Misteli, Tom},<br />doi = {10.1016/j.cell.2010.11.056},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Luco et al. - 2011 - Epigenetics in alternative pre-mRNA splicing(2).pdf:pdf},<br />issn = {1097-4172},<br />journaltitle = {Cell},<br />keywords = {Alternative Splicing,Animals,Chromatin Assembly and Disassembly,Epigenesis, Genetic,Epigenetics on alt splicing,Histones,Histones: metabolism,Humans,RNA Precursors,RNA Precursors: metabolism,Transcription, Genetic},<br />mendeley-tags = {Epigenetics on alt splicing},<br />month = jan,<br />number = {1},<br />pages = {16--26},<br />pmid = {21215366},<br />title = {{Epigenetics in alternative pre-mRNA splicing.}},<br />url = {http://www.sciencedirect.com/science/article/pii/S0092867410013784},<br />volume = {144},<br />year = {2011}<br />}<br /></div></div><div id='Rizzo2012' class='problem severe0'><h2>Rizzo2012 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Standardized collection of MNaseseq experiments enables unbiased dataset comparisons' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Standardized collection of MNaseseq experiments enables unbiased dataset comparisons' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Standardized collection of MNaseseq experiments enables unbiased dataset comparisons' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Standardized collection of MNaseseq experiments enables unbiased dataset comparisons' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Standardized collection of MNaseseq experiments enables unbiased dataset comparisons' target='_blank'>ACM</a></div><div class='reference'>Standardized collection of MNase-seq experiments enables unbiased dataset comparisons.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Rizzo2012,<br />abstract = {BACKGROUND: The organization of eukaryotic DNA into chromatin has a strong influence on the accessibility and regulation of genetic information. The locations and occupancies of a principle component of chromatin, nucleosomes, are typically assayed through use of enzymatic digestion with micrococcal nuclease (MNase). MNase is an endo-exo nuclease that preferentially digests naked DNA and the DNA in linkers between nucleosomes, thus enriching for nucleosome-associated DNA. To determine nucleosome organization genome-wide, DNA remaining from MNase digestion is sequenced using high-throughput sequencing technologies (MNase-seq). Unfortunately, the results of MNase-seq can vary dramatically due to technical differences and this confounds comparisons between MNase-seq experiments, such as examining condition-dependent chromatin organizations.<br />RESULTS: In this study we use MNase digestion simulations to demonstrate how MNase-seq signals can vary for different nucleosome configuration when experiments are performed with different extents of MNase digestion. Signal variation in these simulations reveals an important DNA sampling bias that results from a neighborhood effect of MNase digestion techniques. The presence of this neighborhood effect ultimately confounds comparisons between different MNase-seq experiments. To address this issue we present a standardized chromatin preparation which controls for technical variance between MNase-based chromatin preparations and enables the collection of similarly sampled (matched) chromatin populations. Standardized preparation of chromatin includes a normalization step for DNA input into MNase digestions and close matching of the extent of digestion between each chromatin preparation using gel densitometry analysis. The protocol also includes directions for successful pairing with multiplex sequencing reactions.<br />CONCLUSIONS: We validated our method by comparing the experiment-to-experiment variation between biological replicates of chromatin preparations from S. cerevisiae. Results from our matched preparation consistently produced MNase-seq datasets that were more closely correlated than other unstandardized approaches. Additionally, we validated the ability of our approach at enabling accurate downstream comparisons of chromatin structures, by comparing the specificity of detecting Tup1-dependent chromatin remodeling events in comparisons between matched and un-matched wild-type and tup1$\Delta$ MNase-seq datasets. Our matched MNase-seq datasets demonstrated a significant reduction in non-specific (technical) differences between experiments and were able to maximize the detection of biologically-relevant (Tup1-dependent) changes in chromatin structure.},<br />author = {Rizzo, Jason M and Bard, Jonathan E and Buck, Michael J},<br />doi = {10.1186/1471-2199-13-15},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Rizzo, Bard, Buck - 2012 - Standardized collection of MNase-seq experiments enables unbiased dataset comparisons.pdf:pdf},<br />issn = {1471-2199},<br />journaltitle = {BMC molecular biology},<br />keywords = {Biases on MNase data,Chromatin Assembly and Disassembly,Chromatin Assembly and Disassembly: genetics,DNA,DNA: genetics,DNA: metabolism,Genetic Techniques,High-Throughput Nucleotide Sequencing,Micrococcal Nuclease,Micrococcal Nuclease: diagnostic use,Micrococcal Nuclease: metabolism,Nucleosomes,Nucleosomes: genetics,Reproducibility of Results,Saccharomyces cerevisiae,Saccharomyces cerevisiae: genetics},<br />mendeley-tags = {Biases on MNase data},<br />month = jan,<br />number = {1},<br />pages = {15},<br />pmid = {22559821},<br />title = {{Standardized collection of MNase-seq experiments enables unbiased dataset comparisons.}},<br />url = {http://www.biomedcentral.com/1471-2199/13/15},<br />volume = {13},<br />year = {2012}<br />}<br /></div></div><div id='Stasevich2014' class='problem severe0'><h2>Stasevich2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Regulation of RNA polymerase II activation by histone acetylation in single living cells' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Regulation of RNA polymerase II activation by histone acetylation in single living cells' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Regulation of RNA polymerase II activation by histone acetylation in single living cells' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Regulation of RNA polymerase II activation by histone acetylation in single living cells' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Regulation of RNA polymerase II activation by histone acetylation in single living cells' target='_blank'>ACM</a></div><div class='reference'>Regulation of RNA polymerase II activation by histone acetylation in single living cells</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Stasevich2014,<br />author = {Stasevich, Timothy J. and Hayashi-Takanaka, Yoko and Sato, Yuko and Maehara, Kazumitsu and Ohkawa, Yasuyuki and Sakata-Sogawa, Kumiko and Tokunaga, Makio and Nagase, Takahiro and Nozaki, Naohito and McNally, James G. and Kimura, Hiroshi},<br />doi = {10.1038/nature13714},<br />issn = {0028-0836},<br />journaltitle = {Nature},<br />keywords = {H3K27Ac as activator},<br />mendeley-tags = {H3K27Ac as activator},<br />month = sep,<br />number = {7530},<br />pages = {272--275},<br />publisher = {Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.},<br />shorttitle = {Nature},<br />title = {{Regulation of RNA polymerase II activation by histone acetylation in single living cells}},<br />url = {http://dx.doi.org/10.1038/nature13714},<br />volume = {516},<br />year = {2014}<br />}<br /></div></div><div id='Struhl2013' class='problem severe0'><h2>Struhl2013 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Determinants of nucleosome positioning' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Determinants of nucleosome positioning' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Determinants of nucleosome positioning' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Determinants of nucleosome positioning' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Determinants of nucleosome positioning' target='_blank'>ACM</a></div><div class='reference'>Determinants of nucleosome positioning.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Struhl2013,<br />abstract = {Nucleosome positioning is critical for gene expression and most DNA-related processes. Here we review the dominant patterns of nucleosome positioning that have been observed and summarize the current understanding of their underlying determinants. The genome-wide pattern of nucleosome positioning is determined by the combination of DNA sequence, ATP-dependent nucleosome remodeling enzymes and transcription factors that include activators, components of the preinitiation complex and elongating RNA polymerase II. These determinants influence each other such that the resulting nucleosome positioning patterns are likely to differ among genes and among cells in a population, with consequent effects on gene expression.},<br />author = {Struhl, Kevin and Segal, Eran},<br />doi = {10.1038/nsmb.2506},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Struhl, Segal - 2013 - Determinants of nucleosome positioning.pdf:pdf},<br />issn = {1545-9985},<br />journaltitle = {Nature structural \& molecular biology},<br />keywords = {Adenosine Triphosphate,Adenosine Triphosphate: metabolism,Base Sequence,Chromatin Assembly and Disassembly,DNA-Binding Proteins,DNA-Binding Proteins: genetics,DNA-Binding Proteins: metabolism,Nucleosomes,Nucleosomes: genetics,Nucleosomes: metabolism,Poly dA-dT,RNA Polymerase II,RNA Polymerase II: metabolism,Review of nucleosome positioning,Transcription Factors,Transcription Factors: genetics,Transcription Factors: metabolism,Yeasts,Yeasts: genetics,Yeasts: metabolism},<br />mendeley-tags = {Review of nucleosome positioning},<br />month = mar,<br />number = {3},<br />pages = {267--73},<br />pmid = {23463311},<br />publisher = {Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.},<br />shorttitle = {Nat Struct Mol Biol},<br />title = {{Determinants of nucleosome positioning.}},<br />url = {http://dx.doi.org/10.1038/nsmb.2506},<br />volume = {20},<br />year = {2013}<br />}<br /></div></div><div id='Teif2012' class='problem severe0'><h2>Teif2012 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Genomewide nucleosome positioning during embryonic stem cell development' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Genomewide nucleosome positioning during embryonic stem cell development' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Genomewide nucleosome positioning during embryonic stem cell development' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Genomewide nucleosome positioning during embryonic stem cell development' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Genomewide nucleosome positioning during embryonic stem cell development' target='_blank'>ACM</a></div><div class='reference'>Genome-wide nucleosome positioning during embryonic stem cell development.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Teif2012,<br />abstract = {We determined genome-wide nucleosome occupancies in mouse embryonic stem cells and their neural progenitor and embryonic fibroblast counterparts to assess features associated with nucleosome positioning during lineage commitment. Cell-type- and protein-specific binding preferences of transcription factors to sites with either low (Myc, Klf4 and Zfx) or high (Nanog, Oct4 and Sox2) nucleosome occupancy as well as complex patterns for CTCF were identified. Nucleosome-depleted regions around transcription start and transcription termination sites were broad and more pronounced for active genes, with distinct patterns for promoters classified according to CpG content or histone methylation marks. Throughout the genome, nucleosome occupancy was correlated with certain histone methylation or acetylation modifications. In addition, the average nucleosome repeat length increased during differentiation by 5-7 base pairs, with local variations for specific regions. Our results reveal regulatory mechanisms of cell differentiation that involve nucleosome repositioning.},<br />author = {Teif, Vladimir B and Vainshtein, Yevhen and Caudron-Herger, Ma\"{\i}wen and Mallm, Jan-Philipp and Marth, Caroline and H\"{o}fer, Thomas and Rippe, Karsten},<br />doi = {10.1038/nsmb.2419},<br />issn = {1545-9985},<br />journaltitle = {Nature structural \& molecular biology},<br />keywords = {Animals,Base Sequence,Cell Differentiation,Cell Differentiation: physiology,Cell Lineage,Cell Lineage: physiology,Chromatin Immunoprecipitation,DNA Methylation,Electrophoresis, Agar Gel,Embryonic Stem Cells,Embryonic Stem Cells: metabolism,Embryonic Stem Cells: physiology,Gene Expression Profiling,Gene Expression Regulation, Developmental,Gene Expression Regulation, Developmental: genetic,Gene Expression Regulation, Developmental: physiol,H3K9Ac,High-Throughput Nucleotide Sequencing,Histones,Histones: metabolism,Mice,Mnase ChIP-seq on E3.5: H3K27Ac,Molecular Sequence Data,Nucleosomes,Nucleosomes: genetics,Nucleosomes: metabolism,RNA-seq,Sequence Alignment,Transcription Factors,Transcription Factors: metabolism,and  H3K9me3},<br />mendeley-tags = {H3K9Ac,Mnase ChIP-seq on E3.5: H3K27Ac,RNA-seq,and  H3K9me3},<br />month = nov,<br />number = {11},<br />pages = {1185--92},<br />pmid = {23085715},<br />publisher = {Nature Publishing Group, a division of Macmillan Publishers Limited. All Rights Reserved.},<br />shorttitle = {Nat Struct Mol Biol},<br />title = {{Genome-wide nucleosome positioning during embryonic stem cell development.}},<br />url = {http://dx.doi.org/10.1038/nsmb.2419},<br />volume = {19},<br />year = {2012}<br />}<br /></div></div><div id='Teif2014' class='problem severe0'><h2>Teif2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Nucleosome repositioning links DNA demethylation and differential CTCF binding during stem cell development' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Nucleosome repositioning links DNA demethylation and differential CTCF binding during stem cell development' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Nucleosome repositioning links DNA demethylation and differential CTCF binding during stem cell development' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Nucleosome repositioning links DNA demethylation and differential CTCF binding during stem cell development' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Nucleosome repositioning links DNA demethylation and differential CTCF binding during stem cell development' target='_blank'>ACM</a></div><div class='reference'>Nucleosome repositioning links DNA (de)methylation and differential CTCF binding during stem cell development.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Teif2014,<br />abstract = {During differentiation of embryonic stem cells, chromatin reorganizes to establish cell type-specific expression programs. Here, we have dissected the linkages between DNA methylation (5mC), hydroxymethylation (5hmC), nucleosome repositioning, and binding of the transcription factor CTCF during this process. By integrating MNase-seq and ChIP-seq experiments in mouse embryonic stem cells (ESC) and their differentiated counterparts with biophysical modeling, we found that the interplay between these factors depends on their genomic context. The mostly unmethylated CpG islands have reduced nucleosome occupancy and are enriched in cell type-independent binding sites for CTCF. The few remaining methylated CpG dinucleotides are preferentially associated with nucleosomes. In contrast, outside of CpG islands most CpGs are methylated, and the average methylation density oscillates so that it is highest in the linker region between nucleosomes. Outside CpG islands, binding of TET1, an enzyme that converts 5mC to 5hmC, is associated with labile, MNase-sensitive nucleosomes. Such nucleosomes are poised for eviction in ESCs and become stably bound in differentiated cells where the TET1 and 5hmC levels go down. This process regulates a class of CTCF binding sites outside CpG islands that are occupied by CTCF in ESCs but lose the protein during differentiation. We rationalize this cell type-dependent targeting of CTCF with a quantitative biophysical model of competitive binding with the histone octamer, depending on the TET1, 5hmC, and 5mC state.},<br />author = {Teif, Vladimir B and Beshnova, Daria A and Vainshtein, Yevhen and Marth, Caroline and Mallm, Jan-Philipp and H\"{o}fer, Thomas and Rippe, Karsten},<br />doi = {10.1101/gr.164418.113},<br />issn = {1549-5469},<br />journaltitle = {Genome research},<br />keywords = {H3K9me3,Nucleosomal ChIP-seq: H3K4me3},<br />mendeley-tags = {H3K9me3,Nucleosomal ChIP-seq: H3K4me3},<br />month = aug,<br />number = {8},<br />pages = {1285--95},<br />pmid = {24812327},<br />title = {{Nucleosome repositioning links DNA (de)methylation and differential CTCF binding during stem cell development.}},<br />url = {http://genome.cshlp.org/content/early/2014/05/08/gr.164418.113.abstract},<br />volume = {24},<br />year = {2014}<br />}<br /></div></div><div id='Voigt2012' class='problem severe0'><h2>Voigt2012 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Asymmetrically modified nucleosomes' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Asymmetrically modified nucleosomes' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Asymmetrically modified nucleosomes' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Asymmetrically modified nucleosomes' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Asymmetrically modified nucleosomes' target='_blank'>ACM</a></div><div class='reference'>Asymmetrically modified nucleosomes.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Voigt2012,<br />abstract = {Mononucleosomes, the basic building blocks of chromatin, contain two copies of each core histone. The associated posttranslational modifications regulate essential chromatin-dependent processes, yet whether each histone copy is identically modified in vivo is unclear. We demonstrate that nucleosomes in embryonic stem cells, fibroblasts, and cancer cells exist in both symmetrically and asymmetrically modified populations for histone H3 lysine 27 di/trimethylation (H3K27me2/3) and H4K20me1. Further, we obtained direct physical evidence for bivalent nucleosomes carrying H3K4me3 or H3K36me3 along with H3K27me3, albeit on opposite H3 tails. Bivalency at target genes was resolved upon differentiation of ES cells. Polycomb repressive complex 2-mediated methylation of H3K27 was inhibited when nucleosomes contain symmetrically, but not asymmetrically, placed H3K4me3 or H3K36me3. These findings uncover a potential mechanism for the incorporation of bivalent features into nucleosomes and demonstrate how asymmetry might set the stage to diversify functional nucleosome states.},<br />author = {Voigt, Philipp and LeRoy, Gary and Drury, William J and Zee, Barry M and Son, Jinsook and Beck, David B and Young, Nicolas L and Garcia, Benjamin A and Reinberg, Danny},<br />doi = {10.1016/j.cell.2012.09.002},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Voigt et al. - 2012 - Asymmetrically modified nucleosomes.pdf:pdf},<br />issn = {1097-4172},<br />journaltitle = {Cell},<br />keywords = {Amino Acid Sequence,Animals,Cell Differentiation,Cell Line,Embryonic Stem Cells,Embryonic Stem Cells: metabolism,Evidence of cohexitence of bivalent nucleosomes,Fibroblasts,Fibroblasts: metabolism,HeLa Cells,Histone Code,Histones,Histones: chemistry,Histones: metabolism,Humans,Mice,Molecular Sequence Data,Nucleosomes,Nucleosomes: metabolism,Polycomb-Group Proteins,Polycomb-Group Proteins: metabolism,Promoter Regions, Genetic,Protein Processing, Post-Translational},<br />mendeley-tags = {Evidence of cohexitence of bivalent nucleosomes},<br />month = sep,<br />number = {1},<br />pages = {181--93},<br />pmid = {23021224},<br />title = {{Asymmetrically modified nucleosomes.}},<br />url = {http://www.sciencedirect.com/science/article/pii/S0092867412010677},<br />volume = {151},<br />year = {2012}<br />}<br /></div></div><div id='Waterston2002' class='problem severe0'><h2>Waterston2002 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Initial sequencing and comparative analysis of the mouse genome' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Initial sequencing and comparative analysis of the mouse genome' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Initial sequencing and comparative analysis of the mouse genome' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Initial sequencing and comparative analysis of the mouse genome' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Initial sequencing and comparative analysis of the mouse genome' target='_blank'>ACM</a></div><div class='reference'>Initial sequencing and comparative analysis of the mouse genome.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Waterston2002,<br />abstract = {The sequence of the mouse genome is a key informational tool for understanding the contents of the human genome and a key experimental tool for biomedical research. Here, we report the results of an international collaboration to produce a high-quality draft sequence of the mouse genome. We also present an initial comparative analysis of the mouse and human genomes, describing some of the insights that can be gleaned from the two sequences. We discuss topics including the analysis of the evolutionary forces shaping the size, structure and sequence of the genomes; the conservation of large-scale synteny across most of the genomes; the much lower extent of sequence orthology covering less than half of the genomes; the proportions of the genomes under selection; the number of protein-coding genes; the expansion of gene families related to reproduction and immunity; the evolution of proteins; and the identification of intraspecies polymorphism.},<br />author = {Waterston, Robert H and Lindblad-Toh, Kerstin and Birney, Ewan and Rogers, Jane and Abril, Josep F and Agarwal, Pankaj and Agarwala, Richa and Ainscough, Rachel and Alexandersson, Marina and An, Peter and Antonarakis, Stylianos E and Attwood, John and Baertsch, Robert and Bailey, Jonathon and Barlow, Karen and Beck, Stephan and Berry, Eric and Birren, Bruce and Bloom, Toby and Bork, Peer and Botcherby, Marc and Bray, Nicolas and Brent, Michael R and Brown, Daniel G and Brown, Stephen D and Bult, Carol and Burton, John and Butler, Jonathan and Campbell, Robert D and Carninci, Piero and Cawley, Simon and Chiaromonte, Francesca and Chinwalla, Asif T and Church, Deanna M and Clamp, Michele and Clee, Christopher and Collins, Francis S and Cook, Lisa L and Copley, Richard R and Coulson, Alan and Couronne, Olivier and Cuff, James and Curwen, Val and Cutts, Tim and Daly, Mark and David, Robert and Davies, Joy and Delehaunty, Kimberly D and Deri, Justin and Dermitzakis, Emmanouil T and Dewey, Colin and Dickens, Nicholas J and Diekhans, Mark and Dodge, Sheila and Dubchak, Inna and Dunn, Diane M and Eddy, Sean R and Elnitski, Laura and Emes, Richard D and Eswara, Pallavi and Eyras, Eduardo and Felsenfeld, Adam and Fewell, Ginger A and Flicek, Paul and Foley, Karen and Frankel, Wayne N and Fulton, Lucinda A and Fulton, Robert S and Furey, Terrence S and Gage, Diane and Gibbs, Richard A and Glusman, Gustavo and Gnerre, Sante and Goldman, Nick and Goodstadt, Leo and Grafham, Darren and Graves, Tina A and Green, Eric D and Gregory, Simon and Guig\'{o}, Roderic and Guyer, Mark and Hardison, Ross C and Haussler, David and Hayashizaki, Yoshihide and Hillier, LaDeana W and Hinrichs, Angela and Hlavina, Wratko and Holzer, Timothy and Hsu, Fan and Hua, Axin and Hubbard, Tim and Hunt, Adrienne and Jackson, Ian and Jaffe, David B and Johnson, L Steven and Jones, Matthew and Jones, Thomas A and Joy, Ann and Kamal, Michael and Karlsson, Elinor K and Karolchik, Donna and Kasprzyk, Arkadiusz and Kawai, Jun and Keibler, Evan and Kells, Cristyn and Kent, W James and Kirby, Andrew and Kolbe, Diana L and Korf, Ian and Kucherlapati, Raju S and Kulbokas, Edward J and Kulp, David and Landers, Tom and Leger, J P and Leonard, Steven and Letunic, Ivica and Levine, Rosie and Li, Jia and Li, Ming and Lloyd, Christine and Lucas, Susan and Ma, Bin and Maglott, Donna R and Mardis, Elaine R and Matthews, Lucy and Mauceli, Evan and Mayer, John H and McCarthy, Megan and McCombie, W Richard and McLaren, Stuart and McLay, Kirsten and McPherson, John D and Meldrim, Jim and Meredith, Beverley and Mesirov, Jill P and Miller, Webb and Miner, Tracie L and Mongin, Emmanuel and Montgomery, Kate T and Morgan, Michael and Mott, Richard and Mullikin, James C and Muzny, Donna M and Nash, William E and Nelson, Joanne O and Nhan, Michael N and Nicol, Robert and Ning, Zemin and Nusbaum, Chad and O'Connor, Michael J and Okazaki, Yasushi and Oliver, Karen and Overton-Larty, Emma and Pachter, Lior and Parra, Gen\'{\i}s and Pepin, Kymberlie H and Peterson, Jane and Pevzner, Pavel and Plumb, Robert and Pohl, Craig S and Poliakov, Alex and Ponce, Tracy C and Ponting, Chris P and Potter, Simon and Quail, Michael and Reymond, Alexandre and Roe, Bruce A and Roskin, Krishna M and Rubin, Edward M and Rust, Alistair G and Santos, Ralph and Sapojnikov, Victor and Schultz, Brian and Schultz, J\"{o}rg and Schwartz, Matthias S and Schwartz, Scott and Scott, Carol and Seaman, Steven and Searle, Steve and Sharpe, Ted and Sheridan, Andrew and Shownkeen, Ratna and Sims, Sarah and Singer, Jonathan B and Slater, Guy and Smit, Arian and Smith, Douglas R and Spencer, Brian and Stabenau, Arne and Stange-Thomann, Nicole and Sugnet, Charles and Suyama, Mikita and Tesler, Glenn and Thompson, Johanna and Torrents, David and Trevaskis, Evanne and Tromp, John and Ucla, Catherine and Ureta-Vidal, Abel and Vinson, Jade P and {Von Niederhausern}, Andrew C and Wade, Claire M and Wall, Melanie and Weber, Ryan J and Weiss, Robert B and Wendl, Michael C and West, Anthony P and Wetterstrand, Kris and Wheeler, Raymond and Whelan, Simon and Wierzbowski, Jamey and Willey, David and Williams, Sophie and Wilson, Richard K and Winter, Eitan and Worley, Kim C and Wyman, Dudley and Yang, Shan and Yang, Shiaw-Pyng and Zdobnov, Evgeny M and Zody, Michael C and Lander, Eric S},<br />doi = {10.1038/nature01262},<br />issn = {0028-0836},<br />journaltitle = {Nature},<br />keywords = {Animals,Base Composition,Chromosomes, Mammalian,Chromosomes, Mammalian: genetics,Conserved Sequence,Conserved Sequence: genetics,CpG Islands,CpG Islands: genetics,Evolution, Molecular,Gene Expression Regulation,Genes,Genes: genetics,Genetic Variation,Genetic Variation: genetics,Genome,Genome, Human,Genomics,Humans,Mice,Mice, Knockout,Mice, Transgenic,Mice: classification,Mice: genetics,Models, Animal,Mouse genome size 2.5 Gb,Multigene Family,Multigene Family: genetics,Mutagenesis,Neoplasms,Neoplasms: genetics,Physical Chromosome Mapping,Proteome,Proteome: genetics,Pseudogenes,Pseudogenes: genetics,Quantitative Trait Loci,Quantitative Trait Loci: genetics,RNA, Untranslated,RNA, Untranslated: genetics,Repetitive Sequences, Nucleic Acid,Repetitive Sequences, Nucleic Acid: genetics,Selection, Genetic,Sequence Analysis, DNA,Sex Chromosomes,Sex Chromosomes: genetics,Species Specificity,Synteny},<br />mendeley-tags = {Mouse genome size 2.5 Gb},<br />month = dec,<br />number = {6915},<br />pages = {520--62},<br />pmid = {12466850},<br />publisher = {Macmillian Magazines Ltd.},<br />shorttitle = {Nature},<br />title = {{Initial sequencing and comparative analysis of the mouse genome.}},<br />url = {http://dx.doi.org/10.1038/nature01262},<br />volume = {420},<br />year = {2002}<br />}<br /></div></div><div id='Zhang2014' class='problem severe0'><h2>Zhang2014 (article)</h2> <div class='links'> <a href='http://scholar.google.de/scholar?hl=en&q=Canonical nucleosome organization at promoters forms during genome activation' target='_blank'>Scholar</a> |  <a href='https://www.google.com/search?q=Canonical nucleosome organization at promoters forms during genome activation' target='_blank'>Google</a> |  <a href='http://dblp.org/search/index.php#query=Canonical nucleosome organization at promoters forms during genome activation' target='_blank'>DBLP</a> |  <a href='http://ieeexplore.ieee.org/search/searchresult.jsp?queryText=Canonical nucleosome organization at promoters forms during genome activation' target='_blank'>IEEE</a> |  <a href='http://dl.acm.org/results.cfm?nquery=Canonical nucleosome organization at promoters forms during genome activation' target='_blank'>ACM</a></div><div class='reference'>Canonical nucleosome organization at promoters forms during genome activation.</div><ul></ul><form class='problem_control'><label>checked</label><input type='checkbox' class='checked'/></form><div class='bibtex_toggle'>Current BibLaTex Entry</div><div class='bibtex'>@article{Zhang2014,<br />abstract = {The organization of nucleosomes influences transcriptional activity by controlling accessibility of DNA binding proteins to the genome. Genome-wide nucleosome binding profiles have identified a canonical nucleosome organization at gene promoters, where arrays of well-positioned nucleosomes emanate from nucleosome-depleted regions. The mechanisms of formation and the function of canonical promoter nucleosome organization remain unclear. Here we analyze the genome-wide location of nucleosomes during zebrafish embryogenesis and show that well-positioned nucleosome arrays appear on thousands of promoters during the activation of the zygotic genome. The formation of canonical promoter nucleosome organization is independent of DNA sequence preference, transcriptional elongation, and robust RNA polymerase II (Pol II) binding. Instead, canonical promoter nucleosome organization correlates with the presence of histone H3 lysine 4 trimethylation (H3K4me3) and affects future transcriptional activation. These findings reveal that genome activation is central to the organization of nucleosome arrays during early embryogenesis.},<br />author = {Zhang, Yong and Vastenhouw, Nadine L and Feng, Jianxing and Fu, Kai and Wang, Chenfei and Ge, Ying and Pauli, Andrea and van Hummelen, Paul and Schier, Alexander F and Liu, X Shirley},<br />doi = {10.1101/gr.157750.113},<br />file = {:home/marcelo/.local/share/data/Mendeley Ltd./Mendeley Desktop/Downloaded/Zhang et al. - 2014 - Canonical nucleosome organization at promoters forms during genome activation(2).pdf:pdf},<br />issn = {1549-5469},<br />journaltitle = {Genome research},<br />keywords = {Animals,DNA-Binding Proteins,DNA-Binding Proteins: genetics,Embryonic Development,Embryonic Development: genetics,Genome,Histone-Lysine N-Methyltransferase,Histone-Lysine N-Methyltransferase: genetics,Mnase ChIP-seq on zebrafish. Canonical promoter fo,Nucleosomes,Nucleosomes: genetics,Promoter Regions, Genetic,RNA Polymerase II,RNA Polymerase II: genetics,Sequence Analysis, DNA,Transcription, Genetic,Transcriptional Activation,Transcriptional Activation: genetics,Zebrafish},<br />mendeley-tags = {Mnase ChIP-seq on zebrafish. Canonical promoter fo},<br />month = feb,<br />number = {2},<br />pages = {260--6},<br />pmid = {24285721},<br />title = {{Canonical nucleosome organization at promoters forms during genome activation.}},<br />url = {http://genome.cshlp.org/content/24/2/260.short},<br />volume = {24},<br />year = {2014}<br />}<br /></div></div></body></html>