

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Command Line Utilites &mdash; nucChIP 1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="nucChIP 1 documentation" href="index.html"/>
        <link rel="prev" title="Expected number of histone reads using high quality nucleosomes" href="highQual_nuc.html"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="index.html" class="fa fa-home"> nucChIP</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Prerequisites and installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#installation-on-linux">Installation on Linux</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="exampleCase.html">Example case</a><ul>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#data-summary">1. Data summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#nucleosome-positions">Nucleosome positions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#reproducibility">Reproducibility</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#h3k4me3-and-h3k27ac-footprints-are-coherent-with-known-epigenetic-functions">H3K4me3 and H3K27Ac footprints are coherent with known epigenetic functions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#position-specific-nucleosomes-control-gene-expression">Position-specific nucleosomes control gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#there-isn-t-genome-wide-mnase-enrichment-bias-towards-isolated-nucleosomes">There isn&#8217;t genome-wide MNase enrichment bias towards isolated nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#inclusion-of-alternatively-spliced-exons-is-signaled-by-histone-marks">Inclusion of alternatively spliced exons is signaled by histone marks</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#mnase-normalization-of-nucleosomal-chip-seq">MNase normalization of nucleosomal ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#bibliography">Bibliography</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="AS_exons.html">Alternatively spliced exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="countsPerNuc.html">Counts per nucleosome on canonical nucleosomes around AS exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="internalExons.html">Coverage on internal exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="methods.html">Methods</a><ul>
<li class="toctree-l2"><a class="reference internal" href="methods.html#enrichment-of-histone-marks">Enrichment of histone marks</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Command Line Utilites</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#convert-bam-to-bed">Convert BAM to BED</a></li>
<li class="toctree-l2"><a class="reference internal" href="#counts-the-number-of-overlapping-between-two-bed-files">Counts the number of overlapping between two BED files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#generate-coverage-over-nucleosomes">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#count-reads-per-nucleosome">Count reads per nucleosome</a></li>
<li class="toctree-l2"><a class="reference internal" href="#produces-coverage-figures">Produces coverage figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#meges-the-rows-of-two-bed-files">Meges the rows of two BED files</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">nucChIP</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Command Line Utilites</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/commandLineTools.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="command-line-utilites">
<span id="commandlinetools"></span><h1>Command Line Utilites<a class="headerlink" href="#command-line-utilites" title="Permalink to this headline">¶</a></h1>
<div class="section" id="convert-bam-to-bed">
<h2>Convert BAM to BED<a class="headerlink" href="#convert-bam-to-bed" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>usage: bam2bed [-h] [-b BFILE] [-l LTYPE] [-lower LOWER] [-upper UPPER]
               [-e EXTEN] [-t FILETYPE] [-o OFILE]

Transform BAM reads to resemble the true position of a ChIP seq fragment.

optional arguments:
  -h, --help    show this help message and exit
  -b BFILE      BAM file. Aligned single or paried-end ChIP-seq reads
  -l LTYPE      INT. Library type: 0 if library is paired-end, fragment length
                is estimated from data; different from 0 if library is single-
                end, the value assigned here is used as fragment length.
  -lower LOWER  INT. Lower isize of paired-end read. Default: 0
  -upper UPPER  INT. Upper isize of paired-end read. Default: 200
  -e EXTEN      INT. Half length of each read around their fragment midpoing.
  -t FILETYPE   STR. Type of output file: &#39;bam&#39; or &#39;bed&#39;. If chosen &#39;bam&#39;, and
                optional field, &#39;AL:i&#39;, is added containg the average length
                of the fragments.
  -o OFILE      Output file.
</pre></div>
</div>
</div>
<div class="section" id="counts-the-number-of-overlapping-between-two-bed-files">
<h2>Counts the number of overlapping between two BED files<a class="headerlink" href="#counts-the-number-of-overlapping-between-two-bed-files" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: checkMatches [-h] [-f1 FILE1] [-f2 FILE2] [-o OFILE]

Counts the number of overlapping between two BED files.

optional arguments:
  -h, --help  show this help message and exit
  -f1 FILE1   BED enrichment file replicate 1.
  -f2 FILE2   BED enrichment file replicate 2.
  -o OFILE    BED enrichment file with the intersection of replicates 1 and 2.
</pre></div>
</div>
</div>
<div class="section" id="generate-coverage-over-nucleosomes">
<h2>Generate coverage over nucleosomes<a class="headerlink" href="#generate-coverage-over-nucleosomes" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: getAvrCov [-h] [-b BFILE] [-n NFILE] [-f FSIZE] [-w WWIDTH] [-o OFILE]

Generate coverage over nucleosomes.

optional arguments:
  -h, --help  show this help message and exit
  -b BFILE    BAM file.
  -n NFILE    BED file with enriched nucleosomes.
  -f FSIZE    INT. Fragment size of the reads.
  -w WWIDTH   INT. Half width of plotting window.
  -o OFILE    Prefix of output files.
</pre></div>
</div>
</div>
<div class="section" id="count-reads-per-nucleosome">
<h2>Count reads per nucleosome<a class="headerlink" href="#count-reads-per-nucleosome" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: getCounts [-h] [-b BFILE] [-n NFILE] [-pValue PVALUE] [-upper UPPER]
                 [-lower LOWER] [-l LTYPE] [-e EXTEN] [-o OFILE]

Count reads per nucleosome.

optional arguments:
  -h, --help      show this help message and exit
  -b BFILE        BAM file. ChIP-seq reads.
  -n NFILE        BED file. Nucleosome position data, output of Danpos or
                  iNPS.
  -pValue PVALUE  FLOAT file. pValue cut-off of nucleosomes on BED file.
  -upper UPPER    INT. Upper threshold for insert size. Default: 200
  -lower LOWER    INT. Lower threshold for insert size. Default: 0
  -l LTYPE        INT. Library type: if equal to 0, data is assumed to be
                  paired-end and read lengths are estimated from data itself;
                  if different from 0, data is assumed to be single-end and
                  read lengths use this value extend every read.
  -e EXTEN        INT. Half length of reads extended at their midpoint.
  -o OFILE        STR. Name of output file.
</pre></div>
</div>
</div>
<div class="section" id="produces-coverage-figures">
<h2>Produces coverage figures<a class="headerlink" href="#produces-coverage-figures" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: getFigures [-h] [-b BFILE] [-n NFILE] [-f FRAGL] [-hr HALFR]
                  [-hw HALFW] [-o OFILE]

Generate coverage figures.

optional arguments:
  -h, --help  show this help message and exit
  -b BFILE    BAM file.
  -n NFILE    BED file with nucleosome positions.
  -f FRAGL    Fragment length if single-end library. If library is paired-end,
              provide 0 as fragment length.
  -hr HALFR   Half length of reads extended at their midpoint.
  -hw HALFW   Half length of the windows.
  -o OFILE    Output file.
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h2>Generate coverage over nucleosomes<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: getProfile [-h] [-c CONFFILE] [-d DATABASE] [-w WWIDTH] [-lower LOWER]
                  [-upper UPPER] [-perc PERC] [-plotNucs PLOTNUCS]
                  [-localMaxFile LOCALMAXFILE] [-clustering CLUSTERING]
                  [-t TITLE] [-fs FSIZE] [-kc KCLUSTERS] [-kp KPOINTS]
                  [-cols COLS] [-o PREFIX]

Generate coverage plots.

optional arguments:
  -h, --help            show this help message and exit
  -c CONFFILE           STR. Name of configuration file.
  -d DATABASE           STR. Name of database with TSS info.
  -w WWIDTH             INT. Half width of plotting window.
  -lower LOWER          INT or colon separated list of INT. Lower insert size
                        to be considered. Default: 0
  -upper UPPER          INT. Upper insert size to be considered. Deafult: 200
  -perc PERC            FLOAT. Percentile of top extreme values to be
                        discarded.
  -plotNucs PLOTNUCS    STR. Plot nucleosome positions: &#39;true&#39; or &#39;false&#39;.
                        Default = &#39;false&#39;.
  -localMaxFile LOCALMAXFILE
                        STR. File containing positions of nucleosomes relative
                        to TSS. Default = &#39;none&#39;.
  -clustering CLUSTERING
                        STR. Either kmeans or none.
  -t TITLE              STR. Title of output figure.
  -fs FSIZE             INT. Font size.
  -kc KCLUSTERS         INT. Number of clusters for heatmap&#39;s k-means
                        clustering.
  -kp KPOINTS           INT. Number of points to be used on heatmap&#39;s during
                        k-means clustering.
  -cols COLS            INT. Number of columns for the heatmap.
  -o PREFIX             Prefix of output files.
</pre></div>
</div>
</div>
<div class="section" id="meges-the-rows-of-two-bed-files">
<h2>Meges the rows of two BED files<a class="headerlink" href="#meges-the-rows-of-two-bed-files" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>None
usage: merged [-h] [-f1 F1FILE] [-f2 F2FILE] [-o OFILE]

Merge the rows of two BED files.

optional arguments:
  -h, --help  show this help message and exit
  -f1 F1FILE  BED file. File 1.
  -f2 F2FILE  BED file. File 2.
  -o OFILE    STR. Name of output file.
</pre></div>
</div>
</div>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="highQual_nuc.html" class="btn btn-neutral" title="Expected number of histone reads using high quality nucleosomes"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Marcelo Rivas-Astroza.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>