

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Example case &mdash; nucChIP 1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="nucChIP 1 documentation" href="index.html"/>
        <link rel="next" title="Command Line Utilites" href="commandLineTools.html"/>
        <link rel="prev" title="Prerequisites and installation" href="installation.html"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="index.html" class="fa fa-home"> nucChIP</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Prerequisites and installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#installation-on-linux">Installation on Linux</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Example case</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preprocessing">1. Preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#determining-nucleosome-positions">Determining nucleosome positions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#assigning-histone-marks-to-individual-nucleosomes">Assigning histone marks to individual nucleosomes</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="commandLineTools.html">Command Line Utilites</a><ul>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#convert-bam-to-bed">Convert BAM to BED</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#counts-the-number-of-overlapping-between-two-bed-files">Counts the number of overlapping between two BED files</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#generate-coverage-over-nucleosomes">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#count-reads-per-nucleosome">Count reads per nucleosome</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#produces-coverage-figures">Produces coverage figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#id1">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#meges-the-rows-of-two-bed-files">Meges the rows of two BED files</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#creates-bedgraph-file-with-the-enriched-regions-given-a-signal-and-control-and-expected-counts-files">Creates bedGraph file with the enriched regions given a signal and control, and expected counts files</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">nucChIP</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Example case</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/exampleCase.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="example-case">
<span id="examplecase"></span><h1>Example case<a class="headerlink" href="#example-case" title="Permalink to this headline">¶</a></h1>
<p>To exemplify the use of nucChIP, we&#8217;ll use it to analyze MNChIP-seq data from Rivas-Astroza et al <a class="reference external" href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?token=odsnwwionxcnbqv&amp;acc=GSE73004">(GEO accession number GSM1880581)</a>, and MNase data from Carone et al <a class="reference external" href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1400766">(GEO accession number GSM1400766)</a>. Overall, the procedure is divided in three steps:</p>
<ol class="arabic simple">
<li>Preprocessing: mapping of the MNChIP-seq and MNase-seq reads and data quality assesment.</li>
<li>Creation of nucleosome maps: MNChIP-seq and MNase-seq data are pooled together and use to call nucleosome positions genome-wide.</li>
<li>Tracing histone marks to individual nucleosomes: MNChIP-seq and MNase-seq reads are counted over every nucleosome they happen to overlap, and statistical methods are use to determine whether the enrichment of histone marks is strong enough to call a nuclesome as marked by a histone mark.</li>
</ol>
<p>In what follows, we&#8217;ll describe step-by-step instructions. These instructions are based on a Linux-GNU system.</p>
<div class="section" id="preprocessing">
<h2>1. Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h2>
<p>To store the data we&#8217;ll create the folder <tt class="code bash docutils literal"><span class="pre">data</span></tt>.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$mkdir</span> data
</pre></div>
</div>
<p>and download into it the following data</p>
<ul class="simple">
<li>MNChIP-seq data containing information for four histone marks H3K4me3, H3K27Ac, H3K9me3, and H3K27me3: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?token=odsnwwionxcnbqv&amp;acc=GSE73004">GEO accession number GSM1880581</a>,</li>
<li>MNase data: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1400766">GEO accession number GSM1400766</a>.</li>
</ul>
<p>After unziping the fastq files the <tt class="code bash docutils literal"><span class="pre">data</span></tt> folder should look like this</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nv">$ </span>gunzip * <span class="c"># unziping all files in the current directory</span>
<span class="nv">$ </span>ls data  <span class="c"># list all files in the current directory</span>
H3K4me3_rep1_1.fastq
H3K4me3_rep1_2.fastq
H3K4me3_rep2_1.fastq
H3K4me3_rep2_2.fastq
H3K27Ac_rep1_1.fastq
H3K27Ac_rep1_2.fastq
H3K27Ac_rep2_1.fastq
H3K27Ac_rep2_2.fastq
H3K9me3_1.fastq
H3K9me3_2.fastq
H3K27me3_rep1_1.fastq
H3K27me3_rep1_2.fastq
H3K27me3_rep2_1.fastq
H3K27me3_rep2_2.fastq
</pre></div>
</div>
<p>Then, we&#8217;ll map the file to the mouse genome <a class="reference external" href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm9.zip">mm9</a>  using <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie</a>, and conver the output files from SAM to BAM format using <a class="reference external" href="http://www.htslib.org/">samtools</a>.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># Map libraries with replicates</span>
<span class="k">for </span>library in H3K4me3 H3K27Ac H3K27me3; <span class="k">do</span> <span class="c"># Iterates over all fastq files</span>
<span class="k">for </span>replicate in rep1 rep2; <span class="k">do</span>
<span class="k">   </span><span class="nv">name</span><span class="o">=</span><span class="k">${</span><span class="nv">library</span><span class="k">}</span>_<span class="k">${</span><span class="nv">replicate</span><span class="k">}</span>
   <span class="c"># Map fastq file to mm9</span>
   bowtie2 -x mm9 -1 <span class="k">${</span><span class="nv">name</span><span class="k">}</span>_1.fastq -2 <span class="k">${</span><span class="nv">name</span><span class="k">}</span>_2.fastq -p 10 -s <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam
   <span class="c"># Convert sam to sorted bam file</span>
   samtools view -Sb <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam &gt; <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="c"># Convert sam to bam format</span>
   samtools sort <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sort      <span class="c"># Sort bam file</span>
   rm <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam                  <span class="c"># Remove unsorted bam and sam files</span>
<span class="k">done</span>; <span class="k">done</span>
<span class="c"># Map H3K9me3, the only library without replicates</span>
<span class="nv">name</span><span class="o">=</span>H3K9me3
<span class="c"># Map fastq file to mm9</span>
bowtie2 -x mm9 -1 <span class="k">${</span><span class="nv">name</span><span class="k">}</span>_1.fastq -2 <span class="k">${</span><span class="nv">name</span><span class="k">}</span>_2.fastq -p 10 -s <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam
<span class="c"># Convert sam to sorted bam file</span>
samtools view -Sb <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam &gt; <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="c"># Convert sam to bam format</span>
samtools sort <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sort      <span class="c"># Sort bam file</span>
rm <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.bam <span class="k">${</span><span class="nv">name</span><span class="k">}</span>.sam                  <span class="c"># Remove unsorted bam and sam files</span>
</pre></div>
</div>
<p>To remove duplicates we&#8217;ll use MarkDuplicates from <a class="reference external" href="http://broadinstitute.github.io/picard/">picard tools</a></p>
<div class="highlight-bash"><div class="highlight"><pre><span class="k">for </span>file in *bam; <span class="k">do</span>
<span class="k">   </span><span class="nv">name</span><span class="o">=</span><span class="k">${</span><span class="nv">$file</span><span class="p">%.bam</span><span class="k">}</span>
   java -jar MarkDuplicates.jar <span class="se">\</span>
      <span class="nv">REMOVE_DUPLICATES</span> <span class="o">=</span> <span class="nb">true</span> <span class="se">\</span>
      <span class="nv">ASSUME_SORTED</span>     <span class="o">=</span> <span class="nb">true</span> <span class="se">\</span>
      <span class="nv">METRICS_FILE</span>      <span class="o">=</span> metricFile <span class="se">\</span>
      <span class="nv">INPUT</span>             <span class="o">=</span> <span class="nv">$file</span>.pairend.sort.bam <span class="se">\</span>
      <span class="nv">OUTPUT</span>            <span class="o">=</span> <span class="nv">$file</span>.pairend_rmdup.sort.bam
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="section" id="determining-nucleosome-positions">
<h2>Determining nucleosome positions<a class="headerlink" href="#determining-nucleosome-positions" title="Permalink to this headline">¶</a></h2>
<p>Then we checked that the insert size of the MNChIP-seq and MNase-seq libraries were in the mono-nucleosomeal range: ~ 147 b.  For this we used the tool which generated the following figures</p>
<div class="highlight-bash"><div class="highlight"><pre>fragDistribution <span class="se">\</span>
   -b file.pairend_rmdup.sort.bam <span class="se">\</span>
   -t title_in_figure <span class="se">\</span>
   -bins 30 <span class="se">\</span>
   -o prefix_output_file
</pre></div>
</div>
<p>From Figures ... it can be seen that all MNChIP-seq libraries are in the mono-nucleosomal range. The MNase-seq data, on the other hand, also showed protection of sub-nucleosomal molecules. This was the intended result of by Carone et al protocol in order to explore the positioning of a wider range of molecules. To avoid biasing effect of these reads when discovering nucleosomes we followed Carone et al procedure to discover nucleomes, namely filtering out any reads-pairs with insert sizes outside the 135-165 b range (red vertical lines on Figure ...).</p>
<div class="highlight-bash"><div class="highlight"><pre>bam2bed <span class="se">\</span>
    -b <span class="nv">$bam</span> <span class="se">\</span>
    -l <span class="nv">$fragLength</span> <span class="se">\</span>
    -e 75 <span class="se">\</span>
    -t bed <span class="se">\</span>
    -upper 165 <span class="se">\</span>
    -lower 135 <span class="se">\</span>
    -o <span class="k">${</span><span class="nv">lib</span><span class="k">}</span>.REG.bed
bam2bed <span class="se">\</span>
    -b <span class="nv">$bam</span> <span class="se">\</span>
    -l <span class="nv">$fragLength</span> <span class="se">\</span>
    -e 75 <span class="se">\</span>
    -t bed <span class="se">\</span>
    -o <span class="k">${</span><span class="nv">lib</span><span class="k">}</span>.REG.bed
</pre></div>
</div>
<p>All MNChIP-seq reads plus the filtered MNase-seq reads were pooled into a single library (pooled-dataset) and used to call nucleosomes using the software iNPs.</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="c"># Merge BED files by chromosomes</span>
<span class="k">for </span>chrom in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chrX chrY; <span class="k">do</span>
<span class="k">for </span>lib in H3K4me3_rep1 H3K4me3_rep2 H3K27Ac_rep1 H3K27Ac_rep2 H3K9me3 H3K27me3_rep1 H3K27me3_rep2 mnase; <span class="k">do</span>
   <span class="c"># Print only lines from chromosome chrom</span>
   awk -v <span class="nv">x</span><span class="o">=</span><span class="nv">$chrom</span> <span class="s1">&#39;$1==x&#39;</span> <span class="k">${</span><span class="nv">lib</span><span class="k">}</span>.REG.bed &gt;&gt; <span class="k">${</span><span class="nv">chrom</span><span class="k">}</span>.all.bed
<span class="k">done</span>; <span class="k">done</span>
</pre></div>
</div>
<p>Then we run iNPs</p>
<div class="highlight-bash"><div class="highlight"><pre><span class="nb">declare</span> -A chromSizes
chromSizes<span class="o">[</span><span class="s2">&quot;chr1&quot;</span><span class="o">]=</span>197195432
chromSizes<span class="o">[</span><span class="s2">&quot;chr2&quot;</span><span class="o">]=</span>181748087
chromSizes<span class="o">[</span><span class="s2">&quot;chrX&quot;</span><span class="o">]=</span>166650296
chromSizes<span class="o">[</span><span class="s2">&quot;chr3&quot;</span><span class="o">]=</span>159599783
chromSizes<span class="o">[</span><span class="s2">&quot;chr4&quot;</span><span class="o">]=</span>155630120
chromSizes<span class="o">[</span><span class="s2">&quot;chr5&quot;</span><span class="o">]=</span>152537259
chromSizes<span class="o">[</span><span class="s2">&quot;chr7&quot;</span><span class="o">]=</span>152524553
chromSizes<span class="o">[</span><span class="s2">&quot;chr6&quot;</span><span class="o">]=</span>149517037
chromSizes<span class="o">[</span><span class="s2">&quot;chr8&quot;</span><span class="o">]=</span>131738871
chromSizes<span class="o">[</span><span class="s2">&quot;chr10&quot;</span><span class="o">]=</span>129993255
chromSizes<span class="o">[</span><span class="s2">&quot;chr14&quot;</span><span class="o">]=</span>125194864
chromSizes<span class="o">[</span><span class="s2">&quot;chr9&quot;</span><span class="o">]=</span>124076172
chromSizes<span class="o">[</span><span class="s2">&quot;chr11&quot;</span><span class="o">]=</span>121843856
chromSizes<span class="o">[</span><span class="s2">&quot;chr12&quot;</span><span class="o">]=</span>121257530
chromSizes<span class="o">[</span><span class="s2">&quot;chr13&quot;</span><span class="o">]=</span>120284312
chromSizes<span class="o">[</span><span class="s2">&quot;chr15&quot;</span><span class="o">]=</span>103494974
chromSizes<span class="o">[</span><span class="s2">&quot;chr16&quot;</span><span class="o">]=</span>98319150
chromSizes<span class="o">[</span><span class="s2">&quot;chr17&quot;</span><span class="o">]=</span>95272651
chromSizes<span class="o">[</span><span class="s2">&quot;chr18&quot;</span><span class="o">]=</span>90772031
chromSizes<span class="o">[</span><span class="s2">&quot;chr19&quot;</span><span class="o">]=</span>61342430
chromSizes<span class="o">[</span><span class="s2">&quot;chrY_random&quot;</span><span class="o">]=</span>58682461
chromSizes<span class="o">[</span><span class="s2">&quot;chrY&quot;</span><span class="o">]=</span>15902555
chromSizes<span class="o">[</span><span class="s2">&quot;chrUn_random&quot;</span><span class="o">]=</span>5900358
chromSizes<span class="o">[</span><span class="s2">&quot;chrX_random&quot;</span><span class="o">]=</span>1785075
chromSizes<span class="o">[</span><span class="s2">&quot;chr1_random&quot;</span><span class="o">]=</span>1231697
chromSizes<span class="o">[</span><span class="s2">&quot;chr8_random&quot;</span><span class="o">]=</span>849593
chromSizes<span class="o">[</span><span class="s2">&quot;chr17_random&quot;</span><span class="o">]=</span>628739
chromSizes<span class="o">[</span><span class="s2">&quot;chr9_random&quot;</span><span class="o">]=</span>449403
chromSizes<span class="o">[</span><span class="s2">&quot;chr13_random&quot;</span><span class="o">]=</span>400311
chromSizes<span class="o">[</span><span class="s2">&quot;chr7_random&quot;</span><span class="o">]=</span>362490
chromSizes<span class="o">[</span><span class="s2">&quot;chr5_random&quot;</span><span class="o">]=</span>357350
chromSizes<span class="o">[</span><span class="s2">&quot;chr4_random&quot;</span><span class="o">]=</span>160594
chromSizes<span class="o">[</span><span class="s2">&quot;chr3_random&quot;</span><span class="o">]=</span>41899
chromSizes<span class="o">[</span><span class="s2">&quot;chrM&quot;</span><span class="o">]=</span>16299
chromSizes<span class="o">[</span><span class="s2">&quot;chr16_random&quot;</span><span class="o">]=</span>3994
<span class="c"># Compute nuc locations</span>
<span class="nv">nproc</span><span class="o">=</span>0
<span class="nv">maxProc</span><span class="o">=</span>7
<span class="k">for </span>chrom in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chrX chrY; <span class="k">do</span>
<span class="k">   </span><span class="nv">bedChrom</span><span class="o">=</span><span class="k">${</span><span class="nv">chrom</span><span class="k">}</span>.all.bed
   <span class="nv">libChrom</span><span class="o">=</span><span class="k">${</span><span class="nv">chrom</span><span class="k">}</span>
   <span class="nb">echo</span> <span class="nv">$libChrom</span> <span class="o">==============================================================</span>
   python3 /home/rivasas2/tools/iNPS_V1.0/iNPS_V1.0.py <span class="se">\</span>
       -i <span class="nv">$bedChrom</span> <span class="se">\</span>
       -o <span class="nv">$libChrom</span> <span class="se">\</span>
       -c <span class="nv">$libChrom</span> <span class="se">\</span>
       -l <span class="k">${</span><span class="nv">chromSizes</span><span class="p">[</span><span class="nv">$libChrom</span><span class="p">]</span><span class="k">}</span> &amp;
   <span class="nv">nproc</span><span class="o">=</span><span class="k">$((</span><span class="nv">$nproc</span><span class="o">+</span><span class="m">1</span><span class="k">))</span>
   <span class="k">if</span> <span class="o">[</span> <span class="s2">&quot;$nproc&quot;</span> -ge <span class="s2">&quot;$maxProc&quot;</span> <span class="o">]</span>; <span class="k">then</span>
<span class="k">      </span><span class="nb">wait</span>
<span class="nb">      </span><span class="nv">nproc</span><span class="o">=</span>0
      <span class="nb">echo </span>RESET-------
   <span class="k">fi</span>
<span class="k">done</span>
<span class="c">#######################################################################</span>
<span class="c"># Summarize chr files into one single file</span>
<span class="k">for </span>chrom in chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19 chrX chrY; <span class="k">do</span>
<span class="k">   </span><span class="nv">bedChrom</span><span class="o">=</span><span class="k">${</span><span class="nv">chrom</span><span class="k">}</span>.like_bed
   <span class="nb">echo</span> <span class="nv">$bedChrom</span>
   <span class="k">if</span> <span class="o">[</span> <span class="s2">&quot;${chrom}&quot;</span> <span class="o">==</span> <span class="s2">&quot;chr1&quot;</span> <span class="o">]</span>; <span class="k">then</span>
<span class="k">       </span>cat <span class="nv">$bedChrom</span> &gt; allDataNuc.bed
   <span class="k">else</span> <span class="c"># avoid header lines</span>
       awk <span class="s1">&#39;BEGIN{FS=OFS=&quot;\t&quot;} NR&gt;2&#39;</span> <span class="nv">$bedChrom</span> &gt;&gt; allDataNuc.bed
   <span class="k">fi</span>
<span class="k">done</span>
</pre></div>
</div>
<p>This resulted in 10,292,810 nucleosomes.</p>
</div>
<div class="section" id="assigning-histone-marks-to-individual-nucleosomes">
<h2>Assigning histone marks to individual nucleosomes<a class="headerlink" href="#assigning-histone-marks-to-individual-nucleosomes" title="Permalink to this headline">¶</a></h2>
<p>Once the nucleosomal map is available we&#8217;ll determine whether a nucleosome is marked by a histone mark based on its enrichment of MNChIP-seq reads. We&#8217;ll measure enrichment as the counts of MNChIP-seq reads overlappin a nucleosome. However, care has to be taken to avoid the co-founding effect between the co-localization level of nucleosomes and nucleosomal enrichemnt of histone marks. Namely, nucleosomes highly-colocalized have in general higher counts of MNChIP-seq reads. More details on this and to how to avoid it will be cover in the last step (Calling histone marks over nucleosomes). Let&#8217;s start by counting the MNase-seq and MNase-seq reads per nucleosomes.</p>
<div class="section" id="count-mnchip-seq-mnase-seq-reads-per-nucleosome">
<h3>Count MNChIP-seq MNase-seq reads per nucleosome<a class="headerlink" href="#count-mnchip-seq-mnase-seq-reads-per-nucleosome" title="Permalink to this headline">¶</a></h3>
<p>Then, we used our genome-wide nucleosomal map to count, for each nucleosome, the overlapping MNChIP-seq reads.</p>
<div class="highlight-bash"><div class="highlight"><pre>getCounts <span class="se">\</span>
   -b <span class="nv">$bam</span> <span class="se">\</span>
   -n <span class="nv">$nucFile</span> <span class="se">\</span>
   -pValue 0 <span class="se">\</span>
   -l <span class="nv">$fragLength</span> <span class="se">\</span>
   -lower <span class="nv">$lower</span> <span class="se">\</span>
   -upper <span class="nv">$upper</span> <span class="se">\</span>
   -e 75 <span class="se">\</span>
   -o <span class="k">${</span><span class="nv">lib</span><span class="k">}</span>.<span class="k">${</span><span class="nv">nuc</span><span class="k">}</span>.counts.bed &amp;
</pre></div>
</div>
</div>
<div class="section" id="calling-histone-marks-over-nucleosomes">
<h3>Calling histone marks over nucleosomes<a class="headerlink" href="#calling-histone-marks-over-nucleosomes" title="Permalink to this headline">¶</a></h3>
<p>Expected counts were calculated with the R script <a class="reference external" href="https://szbio.ucsd.edu/public/nucChIP/files/exampleCase/expectedCounts.R">analysis</a>:</p>
<div class="highlight-bash"><div class="highlight"><pre>Rscript analysis.R
</pre></div>
</div>
<p>Determining nucleosomes marked by histone marks.</p>
<div class="highlight-bash"><div class="highlight"><pre>getEnrichedRegions <span class="se">\</span>
   -signal <span class="nv">$signal</span> <span class="se">\</span>
   -control <span class="nv">$control</span> <span class="se">\</span>
   -expV <span class="nv">$expV</span> <span class="se">\</span>
   -prefix <span class="nv">$lib</span>
</pre></div>
</div>
</div>
</div>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="commandLineTools.html" class="btn btn-neutral float-right" title="Command Line Utilites"/>Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="installation.html" class="btn btn-neutral" title="Prerequisites and installation"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Marcelo Rivas-Astroza.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>