

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Expected number of histone reads using high quality nucleosomes &mdash; nucChIP 1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="nucChIP 1 documentation" href="index.html"/>
        <link rel="up" title="Methods" href="methods.html"/>
        <link rel="next" title="Command Line Utilites" href="commandLineTools.html"/>
        <link rel="prev" title="Methods" href="methods.html"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="index.html" class="fa fa-home"> nucChIP</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Prerequisites and installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#installation-on-linux">Installation on Linux</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="exampleCase.html">Example case</a><ul>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#data-summary">1. Data summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#nucleosome-positions">Nucleosome positions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#reproducibility">Reproducibility</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#h3k4me3-and-h3k27ac-footprints-are-coherent-with-known-epigenetic-functions">H3K4me3 and H3K27Ac footprints are coherent with known epigenetic functions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#position-specific-nucleosomes-control-gene-expression">Position-specific nucleosomes control gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#there-isn-t-genome-wide-mnase-enrichment-bias-towards-isolated-nucleosomes">There isn&#8217;t genome-wide MNase enrichment bias towards isolated nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#inclusion-of-alternatively-spliced-exons-is-signaled-by-histone-marks">Inclusion of alternatively spliced exons is signaled by histone marks</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#mnase-normalization-of-nucleosomal-chip-seq">MNase normalization of nucleosomal ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#bibliography">Bibliography</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="AS_exons.html">Alternatively spliced exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="countsPerNuc.html">Counts per nucleosome on canonical nucleosomes around AS exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="internalExons.html">Coverage on internal exons</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="methods.html">Methods</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="methods.html#enrichment-of-histone-marks">Enrichment of histone marks</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="commandLineTools.html">Command Line Utilites</a><ul>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#convert-bam-to-bed">Convert BAM to BED</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#counts-the-number-of-overlapping-between-two-bed-files">Counts the number of overlapping between two BED files</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#generate-coverage-over-nucleosomes">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#count-reads-per-nucleosome">Count reads per nucleosome</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#produces-coverage-figures">Produces coverage figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#id1">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#meges-the-rows-of-two-bed-files">Meges the rows of two BED files</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">nucChIP</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
          <li><a href="methods.html">Methods</a> &raquo;</li>
      
    <li>Expected number of histone reads using high quality nucleosomes</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/highQual_nuc.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="expected-number-of-histone-reads-using-high-quality-nucleosomes">
<h1>Expected number of histone reads using high quality nucleosomes<a class="headerlink" href="#expected-number-of-histone-reads-using-high-quality-nucleosomes" title="Permalink to this headline">Â¶</a></h1>
<p>Here, we only used high confidense nucleosomes (iNPs, p-value&lt;=0.05) to compute <span class="math">\(E(X|n_j)\)</span>.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/17_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/17_H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n1_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n1_H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n2_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n2_H3K4me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 1: Expected number of H3K4me3 reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/14_H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/14_H3K27Ac.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/6_H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/6_H3K27Ac.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K27Ac.expectedCounts.svg" width="45%" /></a>
<p>Figure 2: Expected number of H3K27Ac reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/4_H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/4_H3K9me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/9_H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/9_H3K9me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K9me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 3: Expected number of H3K9me3 reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/12_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/12_H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/5_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/5_H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n3_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/highQual_nuc/n3_H3K27me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 4: Expected number of H3K27me3 reads given supporting MNase reads per nucleosome.</p>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="commandLineTools.html" class="btn btn-neutral float-right" title="Command Line Utilites"/>Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="methods.html" class="btn btn-neutral" title="Methods"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Marcelo Rivas-Astroza.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>