

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Methods &mdash; nucChIP 1 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  
    <link rel="top" title="nucChIP 1 documentation" href="index.html"/>
        <link rel="next" title="Expected number of histone reads using high quality nucleosomes" href="highQual_nuc.html"/>
        <link rel="prev" title="Coverage on internal exons" href="internalExons.html"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        <a href="index.html" class="fa fa-home"> nucChIP</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
        
            <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Prerequisites and installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation.html#installation-on-linux">Installation on Linux</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="exampleCase.html">Example case</a><ul>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#data-summary">1. Data summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#nucleosome-positions">Nucleosome positions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#reproducibility">Reproducibility</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#h3k4me3-and-h3k27ac-footprints-are-coherent-with-known-epigenetic-functions">H3K4me3 and H3K27Ac footprints are coherent with known epigenetic functions</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#position-specific-nucleosomes-control-gene-expression">Position-specific nucleosomes control gene expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#there-isn-t-genome-wide-mnase-enrichment-bias-towards-isolated-nucleosomes">There isn&#8217;t genome-wide MNase enrichment bias towards isolated nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#inclusion-of-alternatively-spliced-exons-is-signaled-by-histone-marks">Inclusion of alternatively spliced exons is signaled by histone marks</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#mnase-normalization-of-nucleosomal-chip-seq">MNase normalization of nucleosomal ChIP-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="exampleCase.html#bibliography">Bibliography</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="AS_exons.html">Alternatively spliced exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="countsPerNuc.html">Counts per nucleosome on canonical nucleosomes around AS exons</a></li>
<li class="toctree-l1"><a class="reference internal" href="internalExons.html">Coverage on internal exons</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Methods</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#enrichment-of-histone-marks">Enrichment of histone marks</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="commandLineTools.html">Command Line Utilites</a><ul>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#convert-bam-to-bed">Convert BAM to BED</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#counts-the-number-of-overlapping-between-two-bed-files">Counts the number of overlapping between two BED files</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#generate-coverage-over-nucleosomes">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#count-reads-per-nucleosome">Count reads per nucleosome</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#produces-coverage-figures">Produces coverage figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#id1">Generate coverage over nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="commandLineTools.html#meges-the-rows-of-two-bed-files">Meges the rows of two BED files</a></li>
</ul>
</li>
</ul>

        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">nucChIP</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Methods</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/methods.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            
  <div class="section" id="methods">
<h1>Methods<a class="headerlink" href="#methods" title="Permalink to this headline">¶</a></h1>
<div class="section" id="enrichment-of-histone-marks">
<h2>Enrichment of histone marks<a class="headerlink" href="#enrichment-of-histone-marks" title="Permalink to this headline">¶</a></h2>
<p>The number of MNase reads supporting the position of a nucleosome is a confounding variable for the number of histone reads overlapping the same nucleosome. This may be due to position-specific properties of the genome, such as chromatin compactness or GC content, that results in local under-sampling of both MNAse and MNase ChIP-seq fragments. This process may introduced spurious signals of histone enrichment as some region may appear highly enriched compared with &#8211;otherwise equal&#8211; under-sampled regions. A widely used correction method is to normalized the number of histone reads by the number of MNase reads. That&#8217;s, on any given nucleosome, <span class="math">\(j\)</span>, the number of histone reads, <span class="math">\(x_j\)</span>, is normalized as the ratio:</p>
<div class="math">
\[r_j = \frac{ x_j } { n_j }\]</div>
<p>where <span class="math">\(n_j\)</span> is the number of MNase reads.</p>
<p>This method assumes that <span class="math">\(x_j\)</span> is linearly dependent on <span class="math">\(n_j\)</span>. However, this may not be a accurate representation of their relation. In particular, is plausible to assume that the proportion of under-sampled genomic regions is more prevalent among nucleosomes with low values of <span class="math">\(n_j\)</span> than among nucleosomes with large values of <span class="math">\(n_j\)</span>. Under this assumption, we hypothesized that the expected value of <span class="math">\(x_j\)</span> should follow a rather saturation curve with respect to <span class="math">\(n_j\)</span>.</p>
<p>To take this effect into account, we normalized <span class="math">\(x_j\)</span> as:</p>
<div class="math">
\[r_j = \frac{ x_j } { E(X|n_j)}\]</div>
<p>where, <span class="math">\(E(X|n_j)\)</span> is the expected number of histone reads per nucleosome, <span class="math">\(X\)</span>, given <span class="math">\(n_j\)</span> supporting MNase reads.</p>
<p>To test whether our normalization method is better than the classical one, we first check our hypothesis by computing <span class="math">\(E(X|n_j)\)</span> for the observed range of <span class="math">\(n_j\)</span> using all genomic nucleosomes (iNPs):</p>
<div class="math">
\[E(X|n_j) = \frac{1}{||J(n_j)||} \sum_{j \in J(n_j)} x_j\]</div>
<p>Here, <span class="math">\(J(n_j)\)</span> is the subset of nucleosomes with <span class="math">\(n_j\)</span> supporting MNase reads.</p>
<p>By compromising all genomic nucleosomes, <span class="math">\(E(X|n_j))\)</span> is not only unbias towards any position-specific biological function, but also tailors 1 as a reference point for <span class="math">\(r_j\)</span>: values of <span class="math">\(r_j\)</span> above and below 1 can be interpreted as enriched or depleted, respectively, of histone marks.</p>
<p>Additionally, since <span class="math">\(E(X|n_j)\)</span> is directly proportional to the total number of reads per histone library, the ratios <span class="math">\(r_j\)</span> are a metric already normalized by library size.</p>
<p>For each MNase ChIP-seq library, after removing outliers (read counts per nucleosome over 99% quantiles) <span class="math">\(E(X|n_j)\)</span> resulted (Figures 1-4) in monotonic transformations of the number of MNase reads per nucleosome. Interestingly, <span class="math">\(E(X|n_j)\)</span> saturates in the direction of the <span class="math">\(x\)</span>-axis, meaning that as the number of supporting of MNase reads gets larger its confounding effect on histone reads gets attenuated. By taking into account this effect, <span class="math">\(E(X|n_j)\)</span> improves the sensitivity of <span class="math">\(r_j\)</span> compared to the widely used method of using <span class="math">\(n_j\)</span> as denominator for <span class="math">\(r_j\)</span>. This difference is specially important among nucleosomes with large values of <span class="math">\(n_j\)</span>, where using <span class="math">\(n_j\)</span> as denominator for <span class="math">\(r_j\)</span> would otherwise over-amplifies the difference between their histone enrichment signals.</p>
<p>These results support our hypothesis, and therefore favor the use of <span class="math">\(E(X|n_j)\)</span> over <span class="math">\(n_j\)</span> as denominators for <span class="math">\(r_j\)</span>.</p>
<p>The linear relationship between <span class="math">\(n_j\)</span> and <span class="math">\(x_j\)</span> would only holds when the proportion of under-sampled nucleosomes remains constant with respect to <span class="math">\(n_j\)</span>. If this may be the case for a particular library, <span class="math">\(E(X|n_j)\)</span> will be simply reduce to <span class="math">\(n_j\)</span>. Thus, <span class="math">\(E(X|n_j)\)</span> can be interpreted as general alternative to <span class="math">\(n_j\)</span> for cases when the proportion of under-sampled nucleosomes doesn&#8217;t remain constant with <span class="math">\(n_j\)</span>.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/17_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/17_H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n1_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n1_H3K4me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n2_H3K4me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n2_H3K4me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 1: Expected number of H3K4me3 reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/14_H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/14_H3K27Ac.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/6_H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/6_H3K27Ac.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K27Ac.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K27Ac.expectedCounts.svg" width="45%" /></a>
<p>Figure 2: Expected number of H3K27Ac reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/4_H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/4_H3K9me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/9_H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/9_H3K9me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K9me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K9me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 3: Expected number of H3K9me3 reads given supporting MNase reads per nucleosome.</p>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/12_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/12_H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/5_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/5_H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/H3K27me3.expectedCounts.svg" width="45%" /></a>
<a class="reference internal image-reference" href="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n3_H3K27me3.expectedCounts.svg"><img src="https://132.239.135.28/public/nucChIP/files/methods/all_nucs/n3_H3K27me3.expectedCounts.svg" width="45%" /></a>
<p>Figure 4: Expected number of H3K27me3 reads given supporting MNase reads per nucleosome.</p>
<p>Results using only high confidense nucleosomes can be seen here:</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="highQual_nuc.html">Expected number of histone reads using high quality nucleosomes</a></li>
</ul>
</div>
</div>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="highQual_nuc.html" class="btn btn-neutral float-right" title="Expected number of histone reads using high quality nucleosomes"/>Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="internalExons.html" class="btn btn-neutral" title="Coverage on internal exons"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2014, Marcelo Rivas-Astroza.
    </p>
  </div>

  <a href="https://github.com/snide/sphinx_rtd_theme">Sphinx theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>